---
title: "Prognostic and biologic relevance of collagen biosynthesis pathway in prostate cancer"
subtitle: "Methods, figures and tables for the transcriptome part"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx

bibliography: coll_biblio.bib

csl: frontiers-in-immunology.csl

header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

library(bookdown)
library(flextable)

knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "png", 
                      dpi = 600, 
                      tab.cap.pre = 'Table ', 
                      tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', font.size = 10)


```

\newpage

# Methods

## Magnetic resonance imaging of the prostate

_Dear co-authors, it's your part as experts:)_

ADC was compared between patient-matched measurements of benign prostate and malignancy by paired T test with Cohen's d effect size statistic.

## Transcriptome analyses

The transcriptome analysis was performed with R version 4.2.3. 
Its details are presented in __Supplementary Methods__.

### Datasets

Five publicly available transcriptome datasets of primary prostate cancer samples, 
the TCGA prostate cancer cohort [@Abeshouse2015] (n = `r study_data$tcga$expression %>% filter(tissue == 'tumor') %>% nrow`), 
GSE16560 (n = `r nrow(study_data$GSE16560$expression)`) [@Sboner2010], 
GSE70768 (n = `r study_data$GSE70768$expression %>% filter(tissue == 'tumor') %>% nrow`) [@Ross-Adams2015], 
GSE70769 (n = `r nrow(study_data$GSE70769$expression)`) [@Ross-Adams2015], and
GSE116918 (n = `r nrow(study_data$GSE116918$expression)`) [@Jain2018],  were re-analyzed. 
Normalized gene expression data were fetched from the GDC data portal for the TCGA or from the Gene Expression Omnibus for the remaining cohorts. 
Donor-matched benign and malignant tissue samples were available for the TCGA 
(n = `r study_data$tcga$expression %>% filter(tissue == 'benign') %>% nrow` specimen pairs) and 
GSE70768 cohort (n = `r study_data$GSE70768$expression %>% filter(tissue == 'benign') %>% nrow`). 
Expression values were $log_2$-transformed prior to analyses. 
Non-immune cell content in cancer samples was estimated with the _MCP-counter_ and _xCell_ algorithms [@Aran2017; @Becht2016].
single sample gene set enrichment scores (ssGSEA) for the Reactome pathway gene signatures retrieved from the MSig database version 7.5.1 were computed with the _GSVA_ algorithm [@Hanzelmann2013].

The collagen pathway genes were retrieved from the published Collagen Signature [@Kocher2021] and the Reactome collagen formation pathway (R-HSA-1474290) and restricted to the gene set present in all investigated datasets (__Supplementary Table S2__).

### Differential expression in benign and malignant prostate tissue

Expression of the collagen pathway gene between patient-matched benign and malignant specimens was compared by the false discovery rate method (FDR) [@Benjamini1995] corrected paired two-tailed T test with Cohen's d effect size statistic [@Cohen2013].

### Collagen clusters

Cancer samples in the training TCGA cohort were clustered in respect to COMBAT-batch adjusted, normalized $log_2$-transformed gene expression levels of the collagen pathway genes with the PAM algorithm with cosine distance between the samples [@Schubert2019; @Leek2012]. 
The algorithm choice was motivated by its high explained clustering variance and stability in 10-fold cross-validation as compared with several other procedures [@Lange2004; @Leng2014]. 
Two sample clusters were defined [@Rousseeuw1987]: collagen^high^ and collagen^low^ named after differences in expression of collagen subunit genes. 
Cancers samples in the GSE16560, GSE70768, GSE70769 and GSE116918 collectives were assigned to the collagen clusters by an inverse distance weighted k-nearest neighbor classifier [@Leng2014].

Expression of the collagen pathway genes in the collagen clusters was compared by FDR-adjusted two-tailed T test with Cohen's d effect size statistic. 
Differences in clinical variables between the collagen clusters were assessed by FDR-corrected Mann-Whitney test with r effect size statistic or FDR-corrected $\chi^2$ test with Cramer V effect size statistic.  

### Gene set variation analysis and differential gene expression in the collagen clusters

ssGSEA scores of the Reactome signatures and $log_2$-transformed gene expression levels were compared between the collagen clusters by FDR-corrected two-tailed T tests with Cohen's d effect size statistic. 
Signatures and genes significantly regulated between the clusters were defined by the significance cutoff (pFDR < 0.05) and at least weak effect size of the differences between the clusters (d $\geq$ 0.2). 

### Activity of signaling pathways and metabolic reactions

Modulation of activity of signaling pathways between the collagen clusters was investigated by the _SPIA_ algorithm. 
Differentially activated or inhibited pathways were identified by the FDR-corrected combined p value for enrichment and pathway perturbation (pGFDR). 
The magnitude of signaling pathway modulation was measured with the tA parameter [@Tarca2009]. 

Modulation of metabolic reactions of the Recon2 human metabolism model based on whole-genome differential gene expression was evaluated with the _BiGGR_ and [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra) R packages as described [@Pichler2023; @Gavai2015; @King2016]. 
Statistical significance of reaction activity estimates in the collagen^high^ cluster as compared with the collagen^low^ subset was assessed by a Monte Carlo simulation with n = 1000 draws from normal distribution of gene expression. 
P values were corrected for multiple testing with the FDR method.

Enrichment of significantly activated or inhibited biochemical reactions within the Recon model metabolic subsystems [@King2016] was investigated by FDR-corrected Fisher's exact test with odds ratio (OR) effect size statistic. 
Significantly activated and inhibited metabolic subsystems were defined by the significance cutoff (pFDR < 0.05) and at least weak effect size of the enrichment (OR $\geq$ 1.44).

### Data and code availability

Publicly available data sets were analyzed. 
Cleared data sets used for analyses will be made available upon request to the corresponding author. 
The transcriptome R analysis pipeline is available from GitHub (https://github.com/PiotrTymoszuk/collagen_pca).

\newpage

# Tables 

```{r tab-cohorts, tab.cap = 'Characteristic of the analyzed cohorts. Numeric variables are presented as medians with interquartile ranges (IQR) and ranges. Qualitative variables are presented as percentages of categories within the complete observation set.'}

flextable::flextable(paper_tbl$cohorts) %>% 
  set_widths(c(3, rep(3.5, 5))) %>% 
  merge_v(1) %>% 
  footnote(1, 1, 
           value = as_paragraph('PSA: prostate-specific antigen.'), 
           part = 'header', 
           ref_symbols = 'a') %>%
  theme_vanilla

```

\newpage

# Figures

```{r fig-norm-tumor, fig.width = figur::convert(paper_fig$norm_tumor, to = 'in')$w, fig.height = figur::convert(paper_fig$norm_tumor, to = 'in')$h, fig.cap = 'Differences in tissue diffusion capacity and expression of the collagen pathway genes between the prostate cancer and benign tissue.'}

paper_fig$norm_tumor$plot

```

__Figure \@ref(fig:fig-norm-tumor). Differences in tissue diffusion capacity and expression of the collagen pathway genes between the prostate cancer and benign tissue.__ 

_(A) Representative MRI axial sequences of the prostate of an 83-year-old patient with prostate cancer (left panel). The T2-weighted images (a) show a 23mm suspicious hypo-intense lesion in the left peripheral zone (red star) of the prostate, which has an early enhancement in the contrast-enhanced T1-weighted image (b). The DWI scan (c) and ADC map (d) show a strong diffusion restriction (black and white character) of the tumor area suggesting a PI-RADS 5 lesion. Statistical significance for differences in diffusion capacity in patient-matched tumor and benign tissue measured by ADC was determined by paired T test with Cohen's d effect size statistic (right panel). Single ADC values are visualized as points, grey lines connect measurements of the same donors. The number of measurement pairs, effect size and p value are displayed in the plot caption._

_(B) Differences in $log_2$-transformed expression levels of `r nrow(globals$genes_interest)` genes related to collagen metabolism between donor-matched tumor and benign prostate tissue were investigated by paired T test with Cohen's d effect size statistic in the GSE70768 and TCGA cohort. P values were corrected for multiple testing with the false discovery method. Normalized mean $log_2$-transformed expression levels in benign and cancer tissue for `r length(unique(unlist(norm_tumor$common)))` genes significantly regulated in both cohorts are presented in a heat map. Numbers of tissue pairs are displayed in the plot caption._

\newpage

```{r fig-coll-clusters, fig.width = figur::convert(paper_fig$coll_clusters, to = 'in')$w, fig.height = figur::convert(paper_fig$coll_clusters, to = 'in')$h, fig.cap = 'Collagen clusters of prostate carcinoma and their clinical characteristic.'}

paper_fig$coll_clusters$plot

```

__Figure \@ref(fig:fig-coll-clusters). Collagen clusters of prostate carcinoma and their clinical characteristic.__ 

_Prostate cancers in the training TCGA cohort were clustered in respect to normalized $log_2$-transformed expression levels of the collagen pathway genes of interest by the PAM algorithm with cosine distance between the observations. Two clusters, collagen^low^ (low) and collagen^high^ (hi) were defined. Cancer samples from the test cohorts (`r globals$study_labels[globals$study_labels != 'TCGA'] %>% paste(collapse = ', ')`) were assigned to the collagen clusters with an inverse distance weighted k-nearest neighbor classifier._

_(A) UMAP (uniform manifold approximation and projection) of the normalized $log_2$-transformed dataset of the collagen gene expression in the training TCGA cohort. Single cancer samples are visualized as points, the collagen cluster assignment is color coded. Numbers of samples in the clusters are indicated in the plot legend._

_(B) Mean normalized $log_2$-transformed expression levels of the collagen pathway genes in the collagen clusters and cohorts presented in a heat map. ECM: extracellular matrix, Pro: proline turnover, Adh. adhesion._

_(C, D) Distribution of Gleason scores (GS, C) and pathological tumor stages (pT, D) in the collagen clusters. Statistical significance was determined by $\chi^2$ test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Gleason score and tumor stage distributions are shown in stack plots. Effect sizes and p values are displayed in the plot captions. Numbers of observations in the clusters are indicated in the X axes._

\newpage

```{r fig-cluster-biology, fig.width = figur::convert(paper_fig$cluster_biology, to = 'in')$w, fig.height = figur::convert(paper_fig$cluster_biology, to = 'in')$h, fig.cap = 'Infiltration, biological processes, signaling and metabolic pathways differentially regulated in the collagen clusters.'}

paper_fig$cluster_biology$plot

```

__Figure \@ref(fig:fig-cluster-biology). Infiltration, biological processes, signaling and metabolic pathways differentially regulated in the collagen clusters.__ 

_(A) Counts of cancer-associated fibroblasts (CAF) and endothelial cells (EC) estimated by the MCP counter algorithm were found to differ significantly between the collagen clusters in all investigated collectives. Statistical significance was assessed by Mann-Whitney test with r effect size statistic. P values were corrected for multiple comparison with the false discovery rate method. Counts of fibroblasts and endothelial cells in the collagen clusters in the TCGA cohort are shown in violin plots. Red diamonds with whiskers represented medians with interquartile ranges. Single tumor samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of observations in the clusters are indicated in the X axes. Full infiltration analysis results are presented in Supplementary Table S5._

_(B) Gene set variation analysis (GSVA) with Reactome pathway gene signatures. Statistical significance for differences in signature single sample gene set enrichment analysis scores (ssGSEA) between the collagen clusters was determined by two-tailed T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Mean ssGSEA scores in the clusters and cohorts for signatures significantly regulated in at least four cohorts with at least weak effect size (d $\geq$ 0.2) are presented in a heat map. Full analysis results are presented in Supplementary Table S7. ECM: extracellular matrix, GF: growth factor, GPCR: G protein-coupled receptor._

_(C) Modulation of KEGG-listed signaling pathways in collagen^high^ (hi) tumors as compared with collagen^low^ cancers was predicted by the SPIA algorithm based on differential gene expression estimates. Predicted magnitude of pathway modulation (tA parameter) for pathways significantly modulated in at least four cohorts was visualized in a bubble plot (tA < 0: inhibition, blue; tA > 0, activation, red, ns: not significant, gray). Points are labeled with their tA values. Full analysis results are presented in Supplementary Table S9. ECM: extracellular matrix, ACT: actin, SCLC: small cell lung carcinoma, S. aureus: Staphylococcus aureus, SLE: systemic lupus erythematosus, TGF-$\beta$: transforming growth factor-beta.._

_(D) Modulation of Recon2 model metabolic reactions in collagen^high^ (hi) tumors as compared with collagen^low^ cancers was predicted by the BiGGR and biggrExtra algorithms based on whole-genome differential gene expression estimates. Enrichment of significantly activated and significantly inhibited reactions in the collagen^high^ within the Recon model metabolic subsystems was investigated by Fisher's exact test. Enrichment p values were corrected for multiple testing with the false discovery rate method. Enrichment odds ratios (OR) for significantly enriched metabolic subsystems with at least weak effect size (OR $\geq$ 1.44)  shared by at least four cohorts are presented in a bubble plot (inhibited: blue, activated: red, not significant: gray). Points are labeled with their odds ratio values. Full analysis results are presented in Supplementary Table S11. Val: valine, Leu: leucine, Ile: isoleucine, FAOX: fatty acid oxidation, PIP: phosphoinositol phosphate._

\newpage

# References