---
title: "Prognostic and biologic relevance of collagen biosynthesis pathway in prostate cancer"
subtitle: "Supplementary material, transcriptome part"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx

bibliography: coll_biblio.bib

csl: frontiers-in-immunology.csl

header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

library(bookdown)
library(flextable)

knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "png", 
                      dpi = 600, 
                      tab.cap.pre = 'Table ', 
                      tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', font.size = 10)


```

\newpage

# Supplementary Tables

```{r tab-norm-tumor, tab.cap = "Expression of the collagen pathway genes in the malignant and benign tissue compared by paired T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. log2-transformed expression values are presented as medians with interquartile ranges (IQR) and ranges. The table is available as a supplementary Excel file."}

suppl_paper_tbl$norm_tumor[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)
  

```

```{r tab-cluster, tab.cap = 'Expression of the cluster-defining collagen pathway genes in the collagen clusters of prostate cancer. Statistical significance was assessed by one-way ANOVA with eta-squared effect size statistic. P values were corrected for multiple testing with the false discovery rate method. log2-transformed expression values are presented as medians with interquartile ranges (IQR) and ranges. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$cluster[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-clinic, tab.cap = 'Clinical characteristic of the collagen clusters. Numeric variables are presented as medians with interquartile ranges (IQR) and ranges. Nominal variables are presented as percentages and counts of categories within the cluster.'}

flextable::flextable(suppl_paper_tbl$clinic) %>% 
  set_widths(c(2.2, 3, rep(3.5, 3), 2.8, 2.8)) %>% 
  merge_v(1:2) %>% 
  footnote(1, 2, 
           value = as_paragraph('PSA: prostate-specific antigen.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  footnote(1, 6:7, 
           value = as_paragraph("Categorical variables: \u03C7\u00B2 test with Cramer's V effec size statistic. Numeric variables: Kruskal-Wallis test with \u03B7\u00B2 effect size statistic. P values corrected for multiple testing with the false discovery rate method."), 
           part = 'header', 
           ref_symbols = 'b') %>% 
  theme_vanilla

```

\newpage

```{r tab-mcp, tab.cap = 'Non-malignant cell numbers predicted for the collagen Non-maignant cell fractions predicted for the collagen Statistical significance was assessed by Kruskal-Wallis test with eta-squared effect size statistic. P values were corrected for multiple testing with the false discovery method. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$mcp[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-xcell, tab.cap = 'clusters by the MCP counter algorithm. clusters by the xCell algorithm. Statistical significance was assessed by Kruskal-Wallis test with eta-squared effect size statistic. P values were corrected for multiple testing with the false discovery method. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$xcell[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-gsva, tab.cap = 'Gene set variation analysis with the Reactome pathway gene signatures. Differences between the collagen intermediate or high clusters versus collagen low cancers were investigated by one-way ANOVA with eta-squared effect size statistic and linear modeling. Results for signatures significantly regulated with moderate-to-large effect size (eta-squared at leat 0.06) in at least four cohorts are presented. P values were corrected for multiple testing with the false discovery rate method (FDR). The table is available as a supplementary Excel file.'}

suppl_paper_tbl$gsva[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-dge, tab.cap = 'Genes differentially expressed in the collagen intermediate or high cluster as compared with collagen low cancers were identified by one-way ANOVA and linear modeling with the 1.25-fold regulation cutoff P values were corrected for multiple testing with the false discovery rate method (FDR). The table is available as a supplementary Excel file.'}

suppl_paper_tbl$dge[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-signaling, tab.cap = 'Signaling pathway activity in the collagen clusters investigated by the SPIA algorithm. Resulat for signaling pathways significantly activated or inhibited in at least four cohorts are shown. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$signaling[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-reactions, tab.cap = 'Biochemical reactions predicted to be significantly activated in the collagen high or collagen low cluster as compared with collagen low cancers. Statistical significance was determined by Monte Carlo simulation. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$reactions[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-subsystems, tab.cap = "Results of enrichment analysis for significantly activated and inhibited biochemical reaction within the Recon metabolism subsystem. Statistical significance was determined by Fisher's exact test corrected for multiple testing with the false discovery rate method (FDR). The table is available as a supplementary Excel file."}

suppl_paper_tbl$subsystems[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

# Supplementary Figures

\newpage

# References