---
title: "Prognostic and biological relevance of collagen biosynthesis pathway in prostate cancer"
subtitle: "Methods, figures and tables for the transcriptome part"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx

bibliography: coll_biblio.bib

csl: frontiers-in-immunology.csl

header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

library(bookdown)
library(flextable)

knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "png", 
                      dpi = 600, 
                      tab.cap.pre = 'Table ', 
                      tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', font.size = 10)


```

\newpage

# Methods

## Magnetic resonance imaging of the prostate

_Dear co-authors, it's your part as experts:)_

ADC was compared between patient-matched measurements of benign prostate and malignancy by paired T test with Cohen's d effect size statistic.

## Transcriptome analyses

The transcriptome analysis was performed with R version 4.2.3. 
Its details are presented in __Supplementary Methods__.

## Transcriptome data sets

Six publicly available transcriptome data sets of primary prostate cancer samples, 
the TCGA prostate cancer cohort [@Liu2018; @Abeshouse2015] 
(n = `r tcga$expression %>% filter(tissue_type == 'tumor') %>% nrow`, RNA sequencing), 
GSE54460 [@Long2014] (n = `r gse54460$expression %>% filter(tissue_type == 'tumor') %>% nrow`, RNA sequencing), 
GSE70768  [@Ross-Adams2015] (n = `r gse70768$expression %>% filter(tissue_type == 'tumor') %>% nrow`, microarray), 
GSE70769  [@Ross-Adams2015] (n = `r gse70769$expression %>% filter(tissue_type == 'tumor') %>% nrow`, microarray), 
GSE220095 [@Schimmelpfennig2023] (n = `r gse220095$expression %>% filter(tissue_type == 'tumor') %>% nrow`, RNA sequencing), 
DKFZ [@Gerhauser2018] (n = `r dkfz$expression %>% filter(tissue_type == 'tumor') %>% nrow`, RNA sequencing) 
were re-analyzed (__Supplementary Table S1__). 
Normalized gene expression data and clinical information were fetched from the cBioportal repository (DKFZ, TCGA) or Gene Expression Omnibus (GEO). 
Selection criteria of the data sets were: availability of Gleason scoring, information of biochemical relapse and relapse-free survival, and availability of expression data for `r nrow(globals$genes_interest)` collagen-related genes covered by proteomic analysis as well (__Supplementary Table S2__).
Donor-matched benign and malignant tissue samples were available for the 
GSE70768 (n = `r norm_tumor$n_numbers$gse70768` tissue pairs) and TCGA cohort (n = `r norm_tumor$n_numbers$tcga` tissue pairs). 
Expression values were $log_2$-transformed prior to analyses. 
Non-immune cell content in cancer samples was estimated with the _MCP Counter_ and _xCell_ algorithms [@Aran2017; @Becht2016].
single sample gene set enrichment scores (ssGSEA) for the Reactome pathway gene signatures retrieved from the MSig database version 7.5.1 were computed with the _GSVA_ algorithm [@Hanzelmann2013].

The collagen-related genes were retrieved from the published Collagen Signature [@Kocher2021] and the Reactome collagen formation pathway (R-HSA-1474290) and restricted to genes covered by the proteomic analyses (__Supplementary Table S2__).

## Differential expression in benign and malignant prostate tissue and cancers stratified by Gleason score

Expression of the collagen-related genes of interest between patient-matched benign and malignant specimens was compared by the false discovery rate method (FDR) corrected paired two-tailed T test with Cohen's d effect size statistic. 
Differentially expressed genes were defined by pFDR < 0.05 and d $\geq$ 0.2 criteria indicative of significant, at least weak expression regulation (__Supplementary Table S3__). 

Differential gene expression in cancer samples stratified by Gleason scoring (5 - 6, 7 and $\geq$ 8) was assessed by FDR-corrected one-way ANOVA with $\eta^2$ effect size statistic. 
Genes with pFDR < 0.05 and $\eta^2 \geq$ 0.02 were deemed differentially regulated (__Supplementary Table S4__). 

## Transcriptomic Collagen Score and prediction of biochemical relapse-free survival

Biochemical relapse-free survival was modeled in the TCGA training cohort as a function of first- and second-order normalized, $\log_2$-transformed expression levels of the collagen-related genes of interest by Elastic Net multi-parameter Cox regression [@Simon2011; @Zou2005]. 
The Transcriptomic Collagen Score was computed as the linear predictor score of the Elastic Net Cox model with non-zero model coefficients presented in __Figure \@ref(fig:fig-coll-score)A__ and __Supplementary Table S15__.

Performance of the Transcriptomic Collagen Score at prediction of biochemical relapse-free survival in the training collective (TCGA) and validation data sets (remaining cohorts) was assessed by univariable Cox regression with Harrell's concordance index [@Harrell1996] and integrated Brier score [@Graf1999] as metrics of goodness of fit and calibration (__Supplementary Table 16__).

## Data and code availability

Publicly available data sets were analyzed. 
Formatted data sets used for analyses will be made available upon request to the corresponding author. 
The transcriptome R analysis pipeline is available from GitHub (https://github.com/PiotrTymoszuk/collagen_pca).

\newpage

# Figures

```{r fig-norm-tumor, fig.width = figur::convert(paper_fig$norm_tumor, to = 'in')$w, fig.height = figur::convert(paper_fig$norm_tumor, to = 'in')$h, fig.cap = 'Differences in tissue diffusion capacity and expression of collagen-related genes between the prostate cancer and benign tissue. Expression of collagen-related genes in cancers stratified by Gleason scores.'}

paper_fig$norm_tumor$plot

```

__Figure \@ref(fig:fig-norm-tumor). Differences in tissue diffusion capacity and expression of collagen-related genes between the prostate cancer and benign tissue. Expression of collagen-related genes in cancers stratified by Gleason scores.__ 

_(A) Representative MRI axial sequences of the prostate of an 83-year-old patient with prostate cancer (left panel). The T2-weighted images (a) show a 23mm suspicious hypo-intense lesion in the left peripheral zone (red star) of the prostate, which has an early enhancement in the contrast-enhanced T1-weighted image (b). The DWI scan (c) and ADC map (d) show a strong diffusion restriction (black and white character) of the tumor area suggesting a PI-RADS 5 lesion. Statistical significance for differences in diffusion capacity in patient-matched tumor and benign tissue measured by ADC was determined by paired T test with Cohen's d effect size statistic (right panel). Single ADC values are visualized as points, grey lines connect measurements of the same donors. The number of analyzed benign - tumor pairs, effect size and p value are displayed in the plot caption._

_(B) Differences in $log_2$-transformed expression of `r nrow(globals$genes_interest)` collagen-related genes between donor-matched tumor and benign prostate tissue samples were investigated by paired T test with Cohen's d effect size statistic in the GSE70768 (n = `r norm_tumor$n_numbers$gse70768`) and TCGA (n = `r norm_tumor$n_numbers$tcga` tissue pairs) cohorts. P values were corrected for multiple testing with the false discovery rate method. Mean normalized $log_2$-transformed expression levels in benign and cancer tissue for `r length(unique(unlist(norm_tumor$common)))` genes significantly regulated in both cohorts with at least weak effect size (pFDR < 0.05, d $\geq$ 0.2) are presented in a heat map._

_(C) Differences in $log_2$-transformed expression levels of `r nrow(globals$genes_interest)` collagen-related genes between cancer samples stratified by Gleason scores were assessed by one-way ANOVA with $\eta^2$ effect size statistic in the GSE54460 (`r format_strata_n(gs_uni$n_numbers$gse54460)`), GSE70768 (`r format_strata_n(gs_uni$n_numbers$gse70768)`), GSE70769 (`r format_strata_n(gs_uni$n_numbers$gse70769)`), GSE220095 (`r format_strata_n(gs_uni$n_numbers$gse220095)`), DKFZ (`r format_strata_n(gs_uni$n_numbers$dkfz)`), and TCGA cohort (`r format_strata_n(gs_uni$n_numbers$tcga)`). Mean normalized $log_2$-transformed expression levels of genes significantly regulated between the Gleason score strata with at least weak effect size (pFDR < 0.05, $\eta^2 \geq$ 0.02) are presented in a heat map._

\newpage

```{r fig-tme, fig.width = figur::convert(paper_fig$tme, to = 'in')$w, fig.height = figur::convert(paper_fig$tme, to = 'in')$h, fig.cap = 'Expression of collagen-related genes in components of the tumor microenvironment.'}

paper_fig$tme$plot

```

__Figure \@ref(fig:fig-tme). Expression of collagen-related genes in components of the tumor microenvironment.__ 

_(A) Cancer samples in six published cohorts were classified as collagen low and collagen high tumors in respect to `r nrow(globals$genes_interest)` collagen-related transcripts by semi-supervised PAM clustering (GSE54460: `r format_strata_n(ana_globals$n_numbers$gse54460)`, GSE70768: `r format_strata_n(ana_globals$n_numbers$gse70768)`, GSE70769: `r format_strata_n(ana_globals$n_numbers$gse70769)`, GSE220095: `r format_strata_n(ana_globals$n_numbers$gse220095)`, DKFZ: `r format_strata_n(ana_globals$n_numbers$dkfz)`, TCGA: `r format_strata_n(ana_globals$n_numbers$tcga)`). Counts of cancer-associated fibroblasts and endothelial cells were predicted by the MCP Counter algorithm and compared between the collagen clusters by Mann-Whitney test with r effect size statistic. Median normalized cell counts with interquartile ranges are presented as boxes, whiskers span over 150% of the interquartile ranges. Values for single cancer samples are depicted as points. All presented effects were significant (pFDR < 0.05) with at least weak (endothelial cells, r > 0.21) or moderate effect size (fibroblasts, r > 0.55)._

_(B) Placeholder for the single cell RNA seq data._

\newpage

```{r fig-coll-score, fig.width = figur::convert(paper_fig$coll_score, to = 'in')$w, fig.height = figur::convert(paper_fig$coll_score, to = 'in')$h, fig.cap = 'Transcriptional Collagen Score and prediction of biochemical relapse-free survival.'}

paper_fig$coll_score$plot

```

__Figure \@ref(fig:fig-coll-score). Transcriptional Collagen Score and prediction of biochemical relapse-free survival.__ 

_Transcriptional Collagen Score was established in the TCGA training cohort by multi-parameter Elastic Net Cox modeling of biochemical relapse-free survival as a function of first- and second-order normalized $log_2$-transformed expression levels of `r nrow(globals$genes_interest)` collagen-related genes._

_(A) Non-zero Elastic Net Cox model coefficients presented as hazard ratios (HR). Point color codes for marker status (blue: favorable, red: unfavorable), point size represents absolute value of HR._

_(B) Performance of the Elastic Net Cox model at prediction of biochemical relapse-free survival in the training TCGA cohort and test collectives assessed by concordance index (C-index, high values indicate high concordance between the predicted and observed survival) and integrated Brier score (IBS, low values indicate good model fit and proper calibration). Values of C-index and IBS expected for random survival prediction are depicted as dashed lines._

_(C) Fractions of biochemical relapse-free patients in low, intermediate and high tertiles of Transcriptional Collagen Score in the training TCGA cohort, and the GSE70768 and DKFZ validation collectives presented in Kaplan-Meier plots. Differences in survival between the tertiles were investigated by Peto-Peto test. Total numbers of patients and numbers of biochemical relapses are displayed in the plot captions. P values are shown in the plots._

\newpage

# References