---
title: "Prognostic and biological relevance of collagen-related genes in prostate cancer"
subtitle: "Supplementary material, transcriptome part"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx

bibliography: coll_biblio.bib

csl: frontiers-in-immunology.csl

header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

library(bookdown)
library(flextable)

knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "png", 
                      dpi = 600, 
                      tab.cap.pre = 'Table ', 
                      tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', font.size = 10)


```

\newpage

# Supplementary Methods

## Software

The analysis was done with R version 4.2.3 (R Foundation). 

Tabular data were handled with the packages _tidyverse_ [@Wickham2019], _rlang_ [@Henry2022] and [_trafo_](https://github.com/PiotrTymoszuk/trafo). 
Text data were handled with _stringi_ [@Gagolewski2021]. 

Import of the TCGA and DKFZ data sets from the cBioportal repository was accomplished with in-house-developed R scripts. 
Transcriptome datasets from the Gene Expression Omnibus were fetched with the _GEOquery_ package [@Sean2007]. 
Gene and probe annotation was accomplished with the _AnnotationDbi_ [@Pages2022] and _org.Hs.eg.db_ packages [@Carlson2022]. 
For prediction of non-malignant cell counts and fractions in cancer samples, the R implementation of the _MCP Counter_ and _xCell_ algorithms provided by the _immunedeconv_ package was used [@Sturm2020; @Aran2017; @Becht2016]. 
Single sample gene set enrichment analysis scores (ssGSEA) were computed with the _GSVA_ algorithm [@Hanzelmann2013] implemented by the package [_gseaTools_](https://github.com/PiotrTymoszuk/gseaTools). 

Semi-supervised clustering was done with the package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools) employing algorithms from the _philentropy_, _cluster_, _factoextra_ and _umap_ packages [@Kassambara2020; @Schubert2019; @Drost2018; @Konopka2022].

For statistical hypothesis testing, effect size calculation, gene set variation analysis (GSVA) and differential gene expression analysis, the packages _rstatix_ [@Kassambara2021], [_ExDA_](https://github.com/PiotrTymoszuk/ExDA) and [_microViz_](https://github.com/PiotrTymoszuk/microViz) were employed. 
Biological process gene ontology (GO) enrichment analysis was performed with the _goana_ algorithm implemented by the _limma_ [@Young2010] and [_microViz_](https://github.com/PiotrTymoszuk/microViz) packages. 
Clustering of GO terms by their semantic similarity was accomplished with _GOSemSim_ [@Yu2010], [_microViz_](https://github.com/PiotrTymoszuk/microViz) and [_clustTools_](https://github.com/PiotrTymoszuk/clustTools). 
Differential modulation of transcriptional regulons and signaling pathways was analyzed with the _collecTRI_ [@Muller-Dott2023] and _PROGENy_ [@Schubert2018] databases by linear modeling tools of the _decoupleR_ package [@Badia-I-Mompel2022]. 
Metabolic reaction activity modeling and metabolic subsystem enrichment analysis were performed with the packages _BiGGR_ [@Gavai2015; @King2016] and [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra). 

Multi-parameter modeling of biochemical relapse-free survival was done with the packages 
_glmnet_ [@Friedman2010] (Ridge, Elastic Net and LASSO cox regression), 
_survivalsvm_ [@Fouodo2018] (Support Vector Machines [SVM]), 
_rfsrc_ [@Ishwaran2008; @Ishwaran2022a] (Random Forest), 
_gbm_ [@Greenwell2022] (Gradient Boosted Machines [GBM]), 
_survival_ [@Therneau2000], 
_surviminer_ [@Kassambara2016] and 
[_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions) were used. 
Univariable analysis of relapse-free survival was accomplished with [_kmOptimizer_](https://github.com/PiotrTymoszuk/kmOptimizer'). 

For visualization of the results, the packages 
_ggplot2_ [@Wickham2019] (scatter, stack, bubble and bar plots, heat maps), 
[_ExDA_](https://github.com/PiotrTymoszuk/ExDA) (violin, stack and ribbon plots), _survminer_ and [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions) (Kaplan-Meier plots), and 
[_microViz_](https://github.com/PiotrTymoszuk/microViz) (heat maps) were used. 
Figures and tables were created with the packages _cowplot_ [@Wilke2019] and _flextable_ [@Gohel2022].

## Data sources and data import

Transcriptome cohorts for analysis in the current report were selected from studies deposited at cBioportal and Gene Expression Omnibus (GEO) with the following criteria: availability of information on Gleason scoring or ISUP risk strata (International Society of Urological Pathology), biochemical relapse and biochemical relapse-free survival, and expression data for `r nrow(globals$genes_interest)` collagen-related genes investigated also in the proteomic part of the project (__Supplementary Table S\@ref(tab:tab-cohorts)__ and __S\@ref(tab:tab-genes)__).

The TCGA prostate cancer data set [@Liu2018; @Abeshouse2015] and DKFZ data set [@Gerhauser2018] consisted of normalized RNA sequencing data for `r tcga$expression %>% filter(tissue_type == 'tumor') %>% nrow` and `r dkfz$expression %>% filter(tissue_type == 'tumor') %>% nrow` cancer samples, respectively, with accompanying clinical information and were obtained from the cBioportal repository with in-house-developed R scripts. 
The TCGA cohort included also 
`r norm_tumor$n_numbers$tcga` donor-matched samples of the benign prostate and prostate cancer tissue.  
The GSE54460 [@Long2014] (RNA sequencing, n = `r gse54460$expression %>% filter(tissue_type == 'tumor') %>% nrow` cancer samples), 
GSE70768 [@Ross-Adams2015] (Illumina HumanHT-12 V4.0 expression beadchip microarray, n = `r gse70768$expression %>% filter(tissue_type == 'tumor') %>% nrow`), 
GSE70769 [@Ross-Adams2015] (Illumina HumanHT-12 V4.0 expression beadchip microarray, n = `r gse70769$expression %>% filter(tissue_type == 'tumor') %>% nrow`), and 
GSE220095 [@Schimmelpfennig2023] (RNA sequencing, n = `r gse220095$expression %>% filter(tissue_type == 'tumor') %>% nrow`) data sets included normalized whole-transcriptome and basic clinical information and were fetched from GEO with the `getGEO()` function from the _GEOquery_ package. 
The GSE70768 data set [@Ross-Adams2015] included also `r norm_tumor$n_numbers$gse70768` donor-matched normal prostate and cancer tissue specimens. 
Demographic and clinical characteristic of the investigated transcriptome data sets is provided in  __Supplementary Table S\@ref(tab:tab-cohorts)__. 

Gene expression levels were transformed with the $log_2(expression)$ and $log_2(count + 1)$ function for the microarray and RNA sequencing data sets prior to further analyses. 
The _MCP Counter_ and _xCell_ estimates of non-malignant cell content in cancer samples were computed with the `deconvolute()` function from the _immunedeconv_ package. 
Gene signatures of the [Reactome pathways](https://reactome.org/) were obtained from the [MSig database](https://www.gsea-msigdb.org/gsea/msigdb/index.jsp), version 7.5.1. 
ssGSEA scores were computed with the `calculate()` function from the [_gseaTools_](https://github.com/PiotrTymoszuk/gseaTools) package. 
ISUP strata were defined based on Gleason Score information as follows: ISUP1: Gleason score 5 - 6, ISUP2: Gleason score 7, ISUP3+: Gleason score $\geq$ 8.

The investigated collagen-related genes included the published Collagen Signature [@Kocher2021] and the Reactome collagen formation pathway genes (R-HSA-1474290) and were constrained to features covered by proteomic analyses of the current report (__Supplementary Table S\@ref(tab:tab-genes)__).

## Statistical hypothesis testing, effect size and statistical significance

Differences in numeric variables between two groups were investigated by paired and unpaired two-tailed T tests with Cohen's d effect size statistic or Mann-Whitney tests with r effect size statistic. 
Differences in numeric variables between three or more groups were assessed by one-way ANOVA with $\eta^2$ effect size statistic. 
Differences in distribution of categorical variables were assessed by $\chi^2$ test with Cramer's V effect size statistic. 
Odds ratio served as an effect size statistic in enrichment analyses. 
Effect sizes were interpreted as follows [@Cohen2013]: 

* Cohen's d, weak: 0.2 - 0.5, moderate: 0.5 - 0.8, large: $\geq$ 0.8

* r, weak: 0.1 - 0.3, moderate: 0.3 - 0.5, large: $\geq$ 0.5

* $\eta^2$: weak: 0.02 - 0.13, moderate: 0.13 - 0.26, large: $\geq$ 0.26

* Cramer's V, weak: 0.1 - 0.3, moderate: 0.3 - 0.5, large: $\geq$ 0.5

* OR, weak: 1.44 - 2.48, moderate: 2.48 - 4.27, large: $\geq$ 4.27

P values were corrected for multiple testing with the false discovery rate method (FDR) [@Benjamini1995] within each analysis step and cohort. 
Effects were considered statistically significant for FDR-corrected p values < 0.05.

## Comparison of gene expression between the benign and cancer tissue, and cancer samples stratified by the ISUP risk scheme

Differences in $log_2$-transformed expression levels of the collagen pathway genes between donor-matched cancer and benign prostate specimens were investigated with paired two-tailed T test with Cohen's effect size statistic in the GSE70768 and TCGA cohorts (__Figure 1B__, __Supplementary Figure S\@ref(fig:fig-norm-tumor)__ and __Supplementary Table S\@ref(tab:tab-norm-tumor)__). 
Differences in gene expression between cancer samples stratified according to the ISUP risk scheme (ISUP1: Gleason Score 5 - 6, ISUP2: Gleason score 7, ISUP3+: Gleason Score $\geq$ 8) were explored by one-way ANOVA with $\eta^2$ effect size statistic (__Figure 1C__, __Supplementary Table S\@ref(tab:tab-gleason)__).
Differentially regulated genes were defined by pFDR < 0.05 and at least weak effect size of the difference (d $\geq$ 0.2 or $\eta^2 \geq$ 0.02). 
The analyses were done with the function `compare_variables()` from the [_ExDA_](https://github.com/PiotrTymoszuk/ExDA) package.

## Collagen clusters of prostate cancer samples

Prostate cancer samples in the TCGA cohort were clustered by normalized $log_2$-transformed expression levels of the collagen-related genes of interest with the PAM (partition around medoids) algorithm with cosine distance between the samples (function `kcluster()`, package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)). 
The choice of the clustering algorithm was motivated by the following criteria (__Supplementary Figure S\@ref(fig:fig-clust-dev)A__): 

* cluster separation measured by mean silhouette statistic [@Rousseeuw1987]

* fraction of potentially misclassified observations (observations with negative silhouette widths) [@Rousseeuw1987]

* explanatory performance gauged by fraction of explained clustering variance (ratio of the total between-cluster sum of squares to the total sum of squares) 

* preservation of the nearest neighborhood (mean fraction of the five nearest neighbors placed in the same cluster) [@Venna2001]

* cluster re-assignment accuracy in 5-fold cross-validation [@Lange2004; @Leng2014]. 
Computation of those numeric statistics of clustering quality in a cross-validation setting was done with the methods `summary()` and `cv()` provided by the [_clustTools_](https://github.com/PiotrTymoszuk/clustTools) package.
The cluster number choice was based on the bend of the peak of the average silhouette statistic [@Rousseeuw1987] (method `plot()`, package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools), __Supplementary Figure S\@ref(fig:fig-clust-dev)B__). 
By this means, two clusters of cancer samples were identified: 'collagen high' and 'collagen low' cancers.

Assignment of cancers samples of the GSE54460, GSE70768, GSE70769, GSE220095, and DKFZ cohorts to the collagen clusters based on normalized $log_2$-transformed expression levels of the collagen-related genes was done with an inverse distance weighted k-nearest neighbor classifier (method `predict()`, package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)) [@Leng2014]. 
The quality of the cluster assignment was assessed by comparing the collagen cluster distribution and numeric statistics of clustering quality described above the training TCGA data set and the test cohorts (__Supplementary Figure S\@ref(fig:fig-clust-dev)CD__). 
In addition, separation of the clusters was assessed by a visual inspection of the UMAP layout plots (method `plot()`, package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)) and heat maps of the mean levels of the clustering factors in the collagen clusters (__Supplementary Figure S\@ref(fig:fig-clusters)__).

Differences in the $log_2$-transformed collagen pathway gene expression between the collagen clusters were investigated by two-tailed T test with Cohen's d effect size statistic (__Supplementary Table S\@ref(tab:tab-cluster)__). 

## Clinical characteristic of the collagen clusters

Differences in numeric clinical variables (age and PSA) between the collagen clusters were assessed by Mann-Whitney test with r effect size statistic (`test_two_groups()`, [_microViz_](https://github.com/PiotrTymoszuk/microViz)).  
Differences in qualitative clinical variables between the collagen clusters were investigated by $\chi^2$ test with Cramer's V effect size statistic (function `compare_variables()`, package [_ExDA_](https://github.com/PiotrTymoszuk/ExDA)). 
The demographic and clinical characteristic of the collagen clusters is presented in __Supplementary Figure S\@ref(fig:fig-clust-tumor)__ and __Supplementary Table S\@ref(tab:tab-clinic)__.

## Non-malignant cell infiltration in the collagen clusters

Non-malignant cell counts and non-malignant cell fractions in the cancer samples were predicted by the _MCP Counter_ [@Becht2016] and _xCell_ [@Aran2017] algorithms, respectively.
Differences in the predicted infiltration levels between the collagen clusters were investigated by Mann-Whitney test with r effect size statistic (function `compare_variables()`, package [_ExDA_](https://github.com/PiotrTymoszuk/ExDA)). 
Populations differing between the clusters with pFDR < 0.05 and at least weak effect size (r $\geq$ 0.1) were considered biologically relevant. 
The analysis results are shown in __Supplementary Figure S\@ref(fig:fig-intiltration)__ and __Supplementary Table S\@ref(tab:tab-infiltration)__.

## Gene set variance analysis of Reactome pathway gene signatures

ssGSEA scores [@Hanzelmann2013] of the Reactome pathway gene signatures were compared between the collagen clusters by two-tailed T test with Cohen's d effect size statistic. 
Signatures found to be significantly regulated (pFDR < 0.05) with at least weak effect size (d $\geq$ 0.2) in at least five cohorts were further investigated. 
To get additional functional insight and for purposes of visualization, these common regulated signatures were classified by their co-expression patterns in the TCGA data set by unsupervised KMEANS clustering (function `kcluster()`, package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)). 
The analysis results are presented in __Supplementary Figure S\@ref(fig:fig-biology)__ and __Supplementary Table S\@ref(tab:tab-gsva)__. 

## Differential gene expression and GO enrichment analysis

Differences in whole-transcriptome gene expression between the collagen clusters were investigated by FDR-corrected two-tailed T test with Cohen's d effect size statistic (function `test_two_groups()`, package [_microViz_](https://github.com/PiotrTymoszuk/microViz)). 
Genes significantly differentially expressed between the clusters were identified by the significance cutoff (pFDR < 0.05) and at least weak effect size of the expression differences (d $\geq$ 0.2). 
Genes found to be differentially regulated between the clusters are listed in __Supplementary Table S\@ref(tab:tab-dge)__.

Biological process GO enrichment within genes upregulated in each of the collagen high and collagen low clusters was analyzed with the _limma_ package _goana_ algorithm [@Young2010], re-implemented by the development package [_microViz_](https://github.com/PiotrTymoszuk/microViz) (function `GOana()`). 
Enrichment p values were adjusted for multiple testing with the FDR method. 
Odds ratio (OR) calculated for a GO term with the following formula served as a measure of effect size:

$$OR_{GO} = \frac{N_{GO|DGE} \times N_{total}}{N_{GO|total} \times N_{DGE}}$$
where $N_{GO|DGE}$ stands for the number of differentially regulated genes assigned to the GO term, 
$N_{total}$ is the total number of investigated genes, 
$N_{GO|total}$ is the number of investigated genes assigned to the GO term, and 
$N_{DGE}$ stands for the number of differentially regulated genes. 
GO terms found to be significantly enriched with at least weak effect size (pFDR < 0.05, OR $\geq$ 1.44) in at least five cohorts were further investigated. 
For additional functional insight, such common enriched GO terms were subjected to multi-dimensional scaling and unsupervised KMEANS clustering with pairwise Wang distance as a measure of semantic similarity (functions `go_sem()` and `kcluster()`, packages _GOSemSim_, [_microViz_](https://github.com/PiotrTymoszuk/microViz) and [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)) [@Yu2010]. 
Results of the GO enrichment analysis are shown in __Supplementary Figure S\@ref(fig:fig-biology)__ and __Supplementary Table S\@ref(tab:tab-go-enrichment)__.

## Transcriptional regulons and signaling pathway activity in the collagen clusters

Modulation of transcriptional regulons, i.e. sets of genes whose expression is controlled by a common transcriptional factor, and activity of selected signaling pathways in collagen high cancers as compared with collagen low tumors were explored by linear modeling with the _collecTRI_ [@Muller-Dott2023] and _PROGENy_ [@Schubert2018] databases, respectively. 
Liner models were constructed for whole-transcriptome effect sizes of gene regulation (Cohen's d, see: __Differential gene expression and GO enrichment analysis__) with the `run_ulm()` and `run_mlm()` tools from the _decoupleR_ package [@Badia-I-Mompel2022]. 
In each case, linear model score (LM score) served as a metric of activity magnitude. 
P values (LM score $\neq$ 0) were corrected for multiple testing with the FDR method. 
Regulons and signaling pathways found to be significantly activated or inhibited in at least five cohorts were further discussed. 
The analysis results are presented in __Supplementary Figure S\@ref(fig:fig-signaling)__,and __Supplementary Tables S\@ref(tab:tab-regulons)__ and __S\@ref(tab:tab-signaling)__.

## Activity of metabolic reactions in the collagen clusters

Rules of assignment of genes to biochemical reactions were retrieved from the Recon2 human metabolism model available via the BiGG database [@King2016] and the R package _BiGGR_ [@Gavai2015]. 
Estimates and standard errors of $log_2$ fold-regulation of expression for all available genes between the collagen clusters were calculated by two-tailed T test as described in __Differential gene expression and GO enrichment analysis__. 
Estimates of biochemical pathway fold-regulation were computed by evaluation of the gene assignment rules in the Recon2 model. 
The 'gene~A~ OR gene~B~' operator was interpreted as arithmetic mean of expression regulation estimates for the genes A and B. 
The 'gene~A~ AND gene~B~' operator was interpreted as minimum of expression regulation estimates for the gene A and gene B [@Gavai2015]. 
Standard deviation, 95% confidence intervals and p values for the predicted reaction regulation estimates were obtained by a Monte Carlo simulation with n = 3000 draws from normal distribution of gene expression regulation estimates (mean: gene expression regulation estimate, standard deviation: standard error of the gene expression regulation estimate) [@Gavai2015; @Pichler2023]. 
P values were corrected for multiple testing with the FDR method. 
The analysis was done with the packages _BiGGR_ [@Gavai2015] and [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra) [@Pichler2023] (function `build_geneSBML()`). 
Biochemical reactions predicted to be significantly activated or inhibited in the collagen high cluster as compared with the collagen low cluster are listed in __Supplementary Table S\@ref(tab:tab-reactions)__.

Enrichment of significantly activated or inhibited biochemical reactions within the Recon model metabolic subsystems [@King2016] was investigated by comparing the frequency of significantly activated or inhibited reactions in the given subsystem with 10000 random samples from the total reaction pool (function `suba()`, package [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra)). 
The magnitude of enrichment was assessed with OR defined for the $i$-th metabolic subsystem as follows: 

$$OR_i = \frac{n_{regulated, i} \times n_{total}}{n_i \times n_{regulated}}$$

where $n_{regulated, i}$ denotes the number of significantly activated or inhibited biochemical reactions within the $i$-th metabolic subsystem, 
$n_{total}$ denotes the total number of investigated reactions, 
$n_i$ denotes the number of metabolic reactions within the $i$-th metabolic subsystem, and 
$n_{regulated}$ is the total number of significantly activated or inhibited biochemical reactions. 
Common significantly activated or inhibited metabolic subsystems were identified as metabolic subsystems significantly activated or inhibited in at least four cohorts, with at least weak effect size defined by OR $\geq$ 1.44 (__Supplementary Figure S\@ref(fig:fig-metabolism)__, __Supplementary Table S\@ref(tab:tab-subsystems)__). 

## Multi-parameter survival modeling with transcriptome data

For survival analyses, $log_2$-transformed expression values of the collagen-related genes (__Supplementary Table S2__) were subjected to batch-adjustment with the ComBat algorithm [@Leek2012]. 
Subsequently, the ComBat-adjusted expression data sets of the GSE54460, GSE70768, GSE70769, GSE220095 cohorts were merged (further referred to as 'pooled GEO' cohort). 

Biochemical relapse-free survival in the pooled GEO training cohort was modeled with normalized (Z-scores) expression values of the collagen-related genes as explanatory variables by the GBM algorithm [@Friedman2001; @Greenwell2022; @Natekin2013; @Friedman2002]. 
Selection of the optimal set of modeling parameters (number of trees, shrinkage factor, interaction depth, and minimal number of observations in the tree node) was accomplished by tuning with 10-fold cross-validation and out-of-fold model deviance as performance statistic (__Supplementary Table S\@ref(tab:tab-tuning)__). 
The tuning was performed with an in-house developed script. 
The GBM model was trained in the pooled GEO cohort with the optimal tuning parameter set with the `gbm()` function (package _gbm_). 
The Transcriptomic Collagen Score was defined as the linear predictor score of the GBM model and was computed for the best model iteration (function `gbm.perf(method = 'cv')`) with the `predict()` method and expression data from the training pooled GEO cohort and the TCGA and DKFZ test collectives. 
Importance of the explanatory variables for the predictive performance of the GBM model in the training pooled GEO cohort were measured with the relative influence method [@Friedman2001] and expressed as gradient of the sum of squared errors attributed to particular variables ($\Delta SSE$, __Figure 3A__). 

Performance of the Transcriptomic Collagen Score at prediction of biochemical relapse-free survival was assessed by uni-variable Cox regression (functions `coxph()` and `as_coxex()`, packages _survival_ and [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions)) [@Therneau2000]. 
The proportional hazard assumption of the Cox models was checked with the `summary(type = 'assumptions')` method, which implements the genuine `cox.zph()` algorithm [@Grambsch1994]. 
Harrell's concordance index [@Harrell1996] and integrated Brier score [@Graf1999] were employed to investigate the overall concordance between the predicted and observed survival and to assess overall calibration (method `summary(type = 'fit')`, package [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions), __Figure 3B__ and __Supplementary Table S\@ref(tab:tab-model-survival)__). 
Of note, as compared with several other machine learning algorithms (Ridge, Elastic Net, and LASSO Cox regression, SVM and Random Forest) [@Friedman2010; @Fouodo2018; @Ishwaran2008; @Ishwaran2022a], the GBM model demonstrated an excellent performance and the highest reproducibility at prediction of biochemical relapse-free survival in the investigated cohorts (__Supplementary Figure S\@ref(fig:fig-surv-algos)__, __Supplementary Table S\@ref(tab:tab-model-survival)__). 
Statistical significance of differences in biochemical relapse-free survival between patients stratified by tertiles of the Transcriptional Collagen Score was determined by Peto-Peto test (functions `surv_fit()` and `surv_pvalue()`, package _survminer_, __Figure 3C__).

In order to assess the confounding effect of single cohorts used to generate the pooled GEO data set on results of modeling of relapse-free survival with the best performing GBM algorithm, three nested Cox proportional hazard model were constructed with the pooled GEO data: (1) a model with cohort as the sole explanatory variable, (2) a model with normalized (Z-score) GBM linear predictor score as the sole explanatory variable, and (3) an additive full mode with cohort and normalized GBM linear predictor score as the explanatory variables. 
Performance of the models was evaluate with C-index and integrated Brier score as described above. 
Hazard ratio estimates for the cohort and GBM predictor terms with 95% confidence intervals were extracted from the models with the `summary()` method from the [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions) package. 
Contribution of the cohort and GBM predictor terms to the additive model were explored by likelihood ratio test (LRT, method `anova()`, package _stats_). 
As presented in __Supplementary Figure S\@ref(fig:fig-surv-cohort)A__, predictive performance of the GBM predictor and the additive Cox model adjusted for the cohort effect was nearly identical. 
Furthermore, the cohort adjustment did not change the hazard ratio estimate of the GBM term in comparison of inference statistics of the GBM predictor-only and the additive Cox model. 
Finally, the highly significant results of the LRT for the GBM predictor term and non-significant LRT results for the cohort term suggest strongly that the GBM predictor but not the cohort effect contribute substantially to the additive model (__Supplementary Figure S\@ref(fig:fig-surv-cohort)B__). 
Collectively, these results let us conclude that the highly predictive results of GBM survival modeling in the GOE pooled cohort are independent from differences in relapse-free survival in single GEO cohorts. 

## Survival analysis for single collagen-related transcripts

The univariable analysis of biochemical relapse-free survival for single collagen-related genes was done in the pooled GEO, TCGA and DKFZ data sets described above for the multi-parameter survival modeling. 
For each of the collagen-related genes, cancer patients were classified as high and low expressors by an expression cutoff corresponding to the largest difference in biochemical relapse-free survival measured by FDR-corrected Mentel-Henszel test (function `find_cutoff()`, package [_kmOptimizer_](https://github.com/PiotrTymoszuk/kmOptimizer), minimal expression strata size set to 25% of the observations). 
Hazard ratio (HR) with 95% confidence intervals for relapse risk in high expressors as compared with low expressors was calculated with the canonical _survival_ package functions _coxph()_, _coef_ and _confint()_. 
Genes associated with unfavorable prognosis were defined by pFDR < 0.05 and HR > 1 for the high vs low expressor comparison. 
Genes associated with favorable prognosis were defined by pFDR < 0.05 and HR < 1 (__Supplementary Figure S\@ref(fig:fig-surv-uni-volcano)__). 
Common survival markers were defined as genes significantly associated with survival in all cohorts (__Supplementary Figure S\@ref(fig:fig-surv-uni-forest)__ - __S\@ref(fig:fig-surv-uni-detal3)__, __Supplementary Table S\@ref(tab:tab-rfs-univariable)__).

## Data and code availability

Publicly available data sets were analyzed. 
Formatted data sets used for analyses will be made available upon request to the corresponding author. 
The transcriptome R analysis pipeline is available from GitHub (https://github.com/PiotrTymoszuk/collagen_pca).

\newpage

# Supplementary Tables

```{r tab-cohorts, tab.cap = 'Characteristic of the analyzed cohorts. Numeric variables are presented as medians with interquartile ranges (IQR) and ranges. Qualitative variables are presented as percentages of categories within the complete observation set.'}

flextable::flextable(suppl_paper_tbl$cohorts) %>% 
  set_widths(c(3, rep(2.2, 6))) %>% 
  merge_v(1) %>% 
  footnote(1, 1, 
           value = as_paragraph('PSA: prostate-specific antigen.'), 
           part = 'header', 
           ref_symbols = 'a') %>%
  theme_vanilla

```

\newpage

```{r tab-genes, tab.cap = 'Collagen-related genes and their classification.'}

flextable::flextable(suppl_paper_tbl$genes) %>% 
  set_widths(widths = c(4.5, 3.5, 3)) %>% 
  italic(j = 2, part = 'body') %>% 
  merge_v(1) %>% 
  theme_vanilla

```

\newpage

```{r tab-norm-tumor, tab.cap = "Expression of the collagen pathway genes in the malignant and benign tissue compared by paired T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. log2-transformed expression values are presented as medians with interquartile ranges (IQR) and ranges. The table is available as a supplementary Excel file."}

suppl_paper_tbl$norm_tumor[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-gleason, tab.cap = 'Expression of the collagen pathway genes in cancer samples stratified by the ISUP risk groups compared by one-way ANOVA with eta-square effect size statistic. P values were corrected for multiple testing with the false discovery rate method. log2-transformed expression values are presented as medians with interquartile ranges (IQR) and ranges. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$gleason[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-cluster, tab.cap = "Expression of the cluster-defining collagen pathway genes in the collagen clusters of prostate cancer. Statistical significance was assessed by two-tailed T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. log2-transformed expression values are presented as medians with interquartile ranges (IQR) and ranges. The table is available as a supplementary Excel file."}

suppl_paper_tbl$cluster[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-clinic, tab.cap = 'Clinical characteristic of the collagen clusters. Numeric variables are presented as medians with interquartile ranges (IQR) and ranges. Nominal variables are presented as percentages and counts of categories within the cluster.'}

flextable::flextable(suppl_paper_tbl$clinic) %>% 
  set_widths(c(2.8, 3.5, 4, 4, 2.8, 2.8)) %>% 
  merge_v(1) %>% 
  footnote(1, 2, 
           value = as_paragraph('PSA: prostate-specific antigen; ISUP: International Society of Urological Pathology risk stratification system.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  footnote(1, 5:6, 
           value = as_paragraph('Qualitative variables: \u03C7\u00B2 test with Cramer V effect size statistic. Numeric variables: Mann-Whitney test with r effect size statistic. P values corrected for multiple testing with the false discovery rate.'), 
           part = 'header', 
           ref_symbols = 'b') %>% 
  theme_vanilla

```

\newpage

```{r tab-infiltration, tab.cap = 'Non-malignant cell numbers predicted for the collagen clusters by the MCP Counter and xCell algorithms. Statistical significance was assessed by Mann-Whitney test with r effect size statistic. P values were corrected for multiple testing with the false discovery method. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$infiltration[1, 1] %>% 
  mutate(Algorithm = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-gsva, tab.cap = "Gene set variation analysis with the Reactome pathway gene signatures. Differences in ssGSEA scores between collagen high and collagen low cancers were investigated by two-tailed T test with Cohen's d effect size statistic. Results for signatures significantly regulated with at least weak effect size (d at least 0.2) in at least five cohorts are presented. P values were corrected for multiple testing with the false discovery rate method (FDR). The table is available as a supplementary Excel file."}

suppl_paper_tbl$reactome[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-dge, tab.cap = "Genes differentially expressed in the collagen high cluster as compared with collagen low cancers were identified by two-tailed T test with the 1.25-fold regulation cutoff and the Cohen's d effect size statistic of 0.2 P values were corrected for multiple testing with the false discovery rate method (FDR). The table is available as a supplementary Excel file."}

suppl_paper_tbl$dge[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-go-enrichment, tab.cap = 'Biological process gene ontology (GO) term enrichment within genes differentially regulated in the collagen clusters. The enrichment analysis was performed with goana tool, enrichment p values were corrected for multiple testing with the false discovery rate (FDR) method. Significant enrichment was defined by pFDR < 0.05 and odds ratio (OR) for enrichment within differentially regulated genes of at least 1.44. OR for enrichment within genes upregulated in the collagen high and collagen low clusters are presented for significant GO terms shared by at least five cohorts. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$go_enrichment[1, 1] %>% 
  mutate(`Collagen cluster` = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-regulons, tab.cap = 'Activity of transcriptional regulons in the collagen high cluster as compared with the collagen low cluster predicted by the collecTRI model. Regulon activity was estimated with uni-parameter linear modeling with whole-transcriptome effect sizes of differential gene expression, p values were corrected for multiple testing with the false discovery rate (FDR) method. Linear model scores are presented for regulons significantly activated or inhibited in at least five cohorts. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$regulons[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-signaling, tab.cap = 'Activity of signaling pathways in the collagen high cluster as compared with the collagen low cluster predicted by the PROGENy model. Pathway activity was estimated with multi-parameter linear modeling with whole-transcriptome effect sizes of differential gene expression. P values were corrected for multiple testing with the false discovery rate (FDR) method,  linear model scores serve as measures of pathway activity. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$signaling[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-reactions, tab.cap = 'Biochemical reactions predicted to be significantly activated in collagen high as compared with collagen low cancers. Statistical significance was determined by a Monte Carlo simulation. P values were corrected for multiple testing with the false discovery rate (FDR) method. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$metabolism[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-subsystems, tab.cap = 'Results of enrichment analysis for significantly activated and inhibited biochemical reactions within the Recon metabolism subsystem. Statistical significance was determined by random sampling from the entire reaction pool p values were corrected for multiple testing with the false discovery rate (FDR) method. Effect size of enrichment of the subsystem in significantly activated or inhibited reactions was measured by odds ratio (OR) statistic. The table is available as a supplementary Excel file.'}

suppl_paper_tbl$subsystems[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r tab-tuning, tab.cap = 'Selection of the optimal parameters of machine learining survival models of biochemical relapse-free survival with expression of the collagen-related genes. The selection process was accomplished by cross-validation tuning in the pooled GEO cohort.'}

flextable::flextable(suppl_paper_tbl$tuning) %>% 
  set_widths(c(3.5, 4, 3, 3)) %>% 
  merge_v(1:2, combine = TRUE) %>% 
  footnote(1, 1, 
           value = as_paragraph('SVM: Support Vector Machines; GBM: Gradient Boosted Machines.'),
           part = 'header', 
           ref_symbols = 'a') %>% 
  footnote(1, 2, 
           value = as_paragraph('CV: cross-validation.'), 
           part = 'header', 
           ref_symbols = 'b') %>% 
  theme_vanilla

```

\newpage

```{r tab-model-survival, tab.cap = 'Performance of machine learning models at prediction of biochemical relapse-free survival with expression of the collagen-related genes.'}

flextable::flextable(suppl_paper_tbl$model_survival) %>% 
  set_widths(c(2.8, 2.7, 2.7, 4.6, 2.5)) %>% 
  merge_v(1:2) %>% 
  footnote(1, 1, 
           value = as_paragraph('SVM: Support Vector Machines; GBM: Gradient Boosted Machines.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  footnote(1, 3, 
           value = as_paragraph('pooled GEO: GSE54460, GSE70768, GSE70769, and GSE220095'), 
           part = 'header', 
           ref_symbols = 'b') %>% 
  theme_vanilla

```

\newpage

```{r tab-rfs-univariable, tab.cap = 'Results of univariable analysis of biochemical relapse-free survival with expression of the collagen-related genes. Prostate cancer patients were stratified by expression cutoffs corresponding to the largest difference in survival assessed by Mentel-Henszel test. Genes found to be significantly associated with the survival in all analyzed cohorts (pooled GEO, TCGA and DKFZ) are presented. The table is available in a supplementary Excel file.'}

suppl_paper_tbl$rfs_univariable[1, 1] %>% 
  mutate( `Association with survival` = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(18)

```

\newpage

# Supplementary Figures 

```{r fig-norm-tumor, fig.width = figur::convert(suppl_paper_fig$norm_tumor, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$norm_tumor, to = 'in')$h, fig.cap = 'Expression of collagen-related genes in the normal prostate and prostate cancer tissue.'}

suppl_paper_fig$norm_tumor$plot

```

__Supplementary Figure S\@ref(fig:fig-norm-tumor). Expression of collagen-related genes in the normal prostate and prostate cancer tissue.__ 

_Differences in $log_2$-transformed expression of `r nrow(globals$genes_interest)` collagen-related genes between donor-matched pairs of the prostate cancer and benign tissue were assessed by paired T test with Cohen's effect size statistic in the GSE70768 and TCGA cohorts. P values were corrected for multiple testing with the false discovery rate (FDR) method. Full analysis results are listed in Supplementary Table S\@ref(tab:tab-norm-tumor)._

_(A) Numbers of significantly up- and downregulated genes in the tumor tissue as compared with the benign tissue in the investigated cohorts presented in Venn plots._

_(B) Mean normalized $log_2$ expression of the collagen-related genes presented as lines. Tinted regions represent the 2 $\times$ SEM (standard error of the mean) intervals. Significant effects are highlighted with bold font in the Y axis. Numbers of tissue pairs are displayed in the plot captions._

\newpage

```{r fig-clust-dev, fig.width = figur::convert(suppl_paper_fig$clust_dev, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$clust_dev, to = 'in')$h, fig.cap = 'Semi-supervised clustering of prostate cancer samples in respect to expression of collagen-related genes.'}

suppl_paper_fig$clust_dev$plot

```

__Supplementary Figure S\@ref(fig:fig-clust-dev). Semi-supervised clustering of prostate cancer samples in respect to expression of collagen-related genes.__ 

_Cancer samples in the TCGA training cohort (n = `r nrow(clust_globals$data$tcga)`) were clustered in respect to normalized, $log_2$-transformed expression levels of the collagen-related genes of interest with the PAM (partition around medoids) algorithm with cosine distance metric. Two clusters were defined: collagen low and collagen high. Assignment of the tumor samples in the training GSE54460 (n = `r nrow(clust_globals$data$gse54460)`), GSE70768 (n = `r nrow(clust_globals$data$gse70768)`), GSE70769 (n = `r nrow(clust_globals$data$gse70769)`), GSE220095 (n = `r nrow(clust_globals$data$gse220095)`), and DKFZ collective (n = `r nrow(clust_globals$data$dkfz)`) to the collagen clusters was accomplished by an inverse distance-weighted k-nearest neighbor classifier._

_(A) Comparison of performance of several clustering algorithms (PAM: partition around medoids, HCL: hierarchical clustering/Ward D2 and KMEANS) and distance metrics (Euclidean, Manhattan, squared Euclidean and cosine) in 5-fold cross-validation of the TCGA data set gauged by mean silhouette width as a measure of cluster separation, fraction of observations with negative silhouette widths indicative of possible misclassification, explained clustering variance and fraction of misclassified 5-nearest neighbors, cluster assignment accuracy in the cross-validation folds. Note the superior performance of the PAM/cosine algorithm used for definition of the collagen clusters._

_(B) Choice of cluster number (k = 2) for PAM/cosine clustering of cancer samples in the TCGA cohort motivated by the peak mean silhouette width statistic._

_(C) Distribution of sizes of the collagen clusters in the TCGA training cohort and the validation collectives. Percentages of samples assigned to the collagen clusters are presented in a stack plot._

_(D) Quality of collagen clusters in the training cohort (TCGA) and validation collectives. Cluster separation was assessed by mean silhouette width, misclassification rate is expressed as fraction of observations with negative silhouette widths. Explanatory performance was gauged by fraction of explained clustering variance. Neighborhood preservation was assessed by mean fraction of 5-nearest neighbors placed in the same cluster._

\newpage

```{r fig-clusters, fig.width = figur::convert(suppl_paper_fig$clusters, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$clusters, to = 'in')$h, fig.cap = 'Expression of the collagen-related genes in the collagen clusters of prostate cancers.'}

suppl_paper_fig$clusters$plot

```

__Supplementary Figure S\@ref(fig:fig-clusters). Expression of the collagen-related genes in the collagen clusters of prostate cancers.__ 

_The collagen clusters of prostate cancer samples were developed by semi-supervised PAM clustering as presented in Supplementary Figure S\@ref(fig:fig-clust-dev). Results of comparison of levels of the cluster-defining factors between the collagen clusters are listed in Supplementary Table S\@ref(tab:tab-cluster)._

_(A) Uniform manifold approximation and projection (UMAP) layout of normalized $log_2$-transformed expression values of the collagen-related genes utilized for definition of the collagen clusters. Each point represents a single cancer sample, cluster assignment is color-coded. Numbers of cancer samples in the collagen clusters are indicated in the figure legends._

_(B) Mean normalized $log_2$-transformed expression values of the cluster-defining collagen-related genes visualized in a heat map. The cluster-defining genes were classified by their biological function (Pro: proline metabolism, ECM: extracellular matrix, Adh.: adhesion)._

\newpage

```{r fig-clust-tumor, fig.width = figur::convert(suppl_paper_fig$clust_tumor, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$clust_tumor, to = 'in')$h, fig.cap = 'ISUP and pathological tumor stage in the collagen clusters.'}

suppl_paper_fig$clust_tumor$plot

```

__Supplementary Figure S\@ref(fig:fig-clust-tumor). ISUP and pathological tumor stage in the collagen clusters.__ 

_Differences in distribution of ISUP risk strata (International Society of Urological Pathology: ISUP1, ISUP2 and ISUP3+) and pathological tumor stages (B) between the collagen clusters were investigated by $\chi^2$ test with Cramer V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Percentages of samples assigned to the ISUP strata or tumor stages within each cluster are shown in stack plots. Effect sizes and p values are displayed in the plot facets. Numbers of complete observations in the clusters are indicated in the Y axes. Full clinical and pathological characteristic of the collagen clusters is shown in Supplementary Table S\@ref(tab:tab-clinic)._

\newpage

```{r fig-intiltration, fig.width = figur::convert(suppl_paper_fig$infiltration, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$infiltration, to = 'in')$h, fig.cap = 'Cancer-associated fibroblasts and endothelial cell infiltration in the collagen clusters.'}

suppl_paper_fig$infiltration$plot

```

__Supplementary Figure S\@ref(fig:fig-intiltration). Cancer-associated fibroblasts and endothelial cell infiltration in the collagen clusters.__ 

_Levels of cancer-associated fibroblasts (A) and endothelial cells (B) in the collagen clusters were predicted by the MCP Counter and xCell algorithms. Statistical significance of differences between the clusters was assessed by Mann-Whitney test with r effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Median infiltration levels with interquartile ranges are visualized as boxes, whiskers span over 150% of the interquartile ranges. Points represent single cancer samples. Effect sizes and p values are displayed in the Y axes, significant effects are highlighted in bold. GSE54460: `r format_strata_n(ana_globals$n_numbers$gse54460)`, GSE70768: `r format_strata_n(ana_globals$n_numbers$gse70768)`, GSE70769: `r format_strata_n(ana_globals$n_numbers$gse70769)`, GSE220095: `r format_strata_n(ana_globals$n_numbers$gse220095)`, DKFZ: `r format_strata_n(ana_globals$n_numbers$dkfz)`, TCGA: `r format_strata_n(ana_globals$n_numbers$tcga)`. Results of the analysis for other cell populations are listed in Supplementary Table S\@ref(tab:tab-infiltration)._

\newpage

```{r fig-biology, fig.width = figur::convert(suppl_paper_fig$biology, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$biology, to = 'in')$h, fig.cap = 'Gene set variation analysis of Reactome pathway gene signatures and gene ontology term enrichment analysis for the collagen clusters.'}

suppl_paper_fig$biology$plot

```

__Supplementary Figure S\@ref(fig:fig-biology). Gene set variation analysis of Reactome pathway gene signatures and gene ontology term enrichment analysis for the collagen clusters.__ 

_(A) Gene set variation analysis (GSVA) with Reactome pathway gene signatures. Statistical significance for differences in signature single sample gene set enrichment analysis scores (ssGSEA) between the collagen clusters was determined by two-tailed T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Mean ssGSEA scores in the clusters and cohorts for signatures significantly regulated in at least five cohorts with at least weak effect size (d $\geq$ 0.2) are presented in a heat map. Full analysis results are presented in Supplementary Table S\@ref(tab:tab-gsva). The common regulated gene signatures were classified in respect their co-expression patterns by KMEANS clustering; signature classification is indicated in the Y axis facets of the heat map (ECM: extracellular matrix, GF: growth factor, GPCR: G protein-coupled receptor)._

_(B) Genes differentially expressed between the collagen clusters were identified by false discovery rate (FDR) corrected two-tailed T test with Cohen's d effect size statistic. Differentially expressed genes were defined by the pFDR < 0.05 and d $\geq$ 0.2 cutoffs (Supplementary Table \@ref(tab:tab-dge)). Biological process gene ontology (GO) term enrichment within genes significantly unregulated in the collagen high and the collagen low clusters was investigated with the goana algorithm. Enrichment p values were adjusted for multiple testing with the FDR method, odds ratio of enrichment over the whole-transcriptome GO frequency served as an effect size statistic. GO terms significantly enriched with at least weak effect size (OR $\geq$ 1.44) in at least five cohorts were subjected to multi-dimensional scaling and unsupervised KMEANS clustering in respect to their semantic similarity. Multi-dimensional scaling layouts of Wang distances between the common enriched GO terms are presented. Each point represents a single GO term, GO cluster assignment is color coded. Complete results of the GO enrichment analysis are listed in Supplementary Table S\@ref(tab:tab-go-enrichment)._

\newpage

```{r fig-signaling, fig.width = figur::convert(suppl_paper_fig$signaling, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$signaling, to = 'in')$h, fig.cap = 'Transcriptional regulons and signaling pathway activity in the collagen clusters.'}

suppl_paper_fig$signaling$plot

```

__Supplementary Figure S\@ref(fig:fig-signaling). Transcriptional regulons and signaling pathway activity in the collagen clusters.__ 

_Differences in activity of transcriptional regulons (gene sets controlled by a common transcription factor, A) and signaling pathways (B) in the collagen high cluster as compared with the collagen low cluster were assessed by linear modeling with the collecTRI and PROGENy databases implemented by the decoupler algorithm. Linear model score (LM score) served as a measure of regulon or pathway activity, p values for significant modulation of activity (LM score $\neq$ 0) were corrected for multiple testing with the false discovery rate method. Full analysis results are listed in Supplementary Tables S\@ref(tab:tab-regulons) and S\@ref(tab:tab-signaling)._

_(A) LM scores of top modulated transcriptional regulons found to be significantly activated or inhibited in at least five cohorts visualized as bubble plots. Point color codes for the regulation sign, point size codes for absolute value of the LM score. LM scores for regulon activity in particular cohorts are presented next to the data points._ 

_(B) LM scores of signaling pathways found to be significantly activated or inhibited in at least five cohorts visualized as bubble plots. Point color codes for the regulation sign, point size codes for absolute value of the LM score. LM scores for regulon activity in particular cohorts are presented next to the data points._ 

\newpage

```{r fig-metabolism, fig.width = figur::convert(suppl_paper_fig$metabolism, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$metabolism, to = 'in')$h, fig.cap = 'Metabolism in the collagen clusters.'}

suppl_paper_fig$metabolism$plot

```

__Supplementary Figure S\@ref(fig:fig-metabolism). Metabolism in the collagen clusters.__ 

_Modulation of Recon2 model metabolic reactions in collagen high tumors as compared with collagen low cancers was predicted by Monte Carlo simulation implemented by the BiGGR and biggrExtra algorithms based on whole-genome differential gene expression estimates (Supplementary Table \@ref(tab:tab-reactions)). Enrichment of significantly activated and significantly inhibited reactions in the Recon metabolic subsystems was assessed by comparing the observed frequency of the subsystem's significantly regulated reactions with 10000 random draws from the entire reaction pool. Enrichment p values were corrected for multiple testing with the false discovery rate method. Enrichment odds ratios (OR) for significantly enriched metabolic subsystems with at least weak effect size (OR $\geq$ 1.44)  shared by at least five cohorts are presented in a bubble plot. Point color codes for reaction modulation status, point sizes represent absolute values of OR. Points are labeled with their OR values. Full analysis results are presented in Supplementary Table S\@ref(tab:tab-subsystems)._

\newpage

```{r fig-surv-algos, fig.width = figur::convert(suppl_paper_fig$surv_algos, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_algos, to = 'in')$h, fig.cap = 'Comparison of performance of various machine learning algorithms at prediction of biochemical relapse-free survival with expression of the collagen-related genes.'}

suppl_paper_fig$surv_algos$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-algos). Comparison of performance of various machine learning algorithms at prediction of biochemical relapse-free survival with expression of the collagen-related genes.__ 

_Biochemical relapse-free survival was modeled with normalized, $log_2$-transformed and ComBat-adjusted expression levels of the collagen-related genes by various machine learning algorithms (Ridge Cox regression, Elastic Net Cox regression, LASSO Cox regression, Support Vector Machines [SVM], Random Forest, and Gradient Boosted Machines [GBM]). Their performance at prediction of biochemical relapse-free survival in the training pooled GEO cohort, and the test TCGA and DKFZ collectives was assessed by Harrell's concordance index (C-index) and integrated Brier scores (IBS). High C-index values indicate good concordance between the observed and predicted survival. Low values of IBS indicate good model calibration._
_C-index and IBS values are presented in scatter plots. Each point represents a single algorithm. Values of C-index and IBS expected for a model predicting survival at random are visualized as dashed lines._

\newpage

```{r fig-surv-cohort, fig.width = figur::convert(suppl_paper_fig$surv_cohort, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_cohort, to = 'in')$h, fig.cap = 'Assessment of the confounding effect of the cohort on results of survival GBM modeling in the pooled GEO data set.'}

suppl_paper_fig$surv_cohort$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-cohort). Assessment of the confounding effect of the cohort on results of survival GBM modeling in the pooled GEO data set.__ 
_To investigate the potentially confounding effect of single cohorts in the pooled GEO data set on results of GBM survival modeling, three Cox models were constructed: a cohort-only model, a model with the GBM linear predictor score as the sole explanatory factor, and an additive model with the cohort and GBM predictor score as explanatory factors._
_(A) Evaluation of predictive performance with the Cox models in the pooled GEO cohort. Concordance between the predicted and observed survival was assessed by concordance index (C-index). Model calibration and forecast quality was measured with integrated Brier score (IBS). Single Cox models are visualized as points. Dashed lines represent values of C-index and IBS expected for a meaningless model._
_(B) Inference of the cohort and GBM predictor terms in the Cox models. Hazard ratios (HR) with are presented with their 95% confidence intervals (CI) in a Forest plot. Results of likelihood ratio test for contribution of the cohort and GBM predictor terms to the additive model are displayed in the plot facets._

\newpage

```{r fig-surv-uni-volcano, fig.width = figur::convert(suppl_paper_fig$surv_uni_volcano, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_uni_volcano, to = 'in')$h, fig.cap = 'Differences in biochemical relapse-free survival between high and low expressors of the collagen-related genes.'}

suppl_paper_fig$surv_uni_volcano$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-uni-volcano). Differences in biochemical relapse-free survival between high and low expressors of the collagen-related genes.__ 

_Cancer patients were stratified by cutoffs of normalized expression for each collagen-related genes which corresponded to the largest differences in biochemical relapse-free survival assessed by false discovery rate (FDR) corrected Mentel-Henszel test into high and low expressors for particular genes. Relative risk of biochemical relapse in high expressors as compared with low expressors was modeled by Cox proportional hazard regression and expressed as hazard ratio (HR). Unfavorable survival markers were defined by pFDR < 0.05 and HR > 1. Favorable survival markers were defined by pFDR < 0.05 and HR < 1. HR and pFDR values are presented in Volcano plots. Each point represents a single collagen-related gene. Significance and sign of association with the relapse risk are color coded. Significant genes are labeled with their symbols. Dashed lines indicate the significance and regulation cutoffs._

\newpage

```{r fig-surv-uni-forest, fig.width = figur::convert(suppl_paper_fig$surv_uni_forest, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_uni_forest, to = 'in')$h, fig.cap = 'Collagen-related transcriptional markers of biochemical relapse risk shared by all investigated cohorts.'}

suppl_paper_fig$surv_uni_forest$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-uni-forest). Collagen-related transcriptional markers of biochemical relapse risk shared by all investigated cohorts.__ 

_Differences in biochemical relapse-free survival in high expressors versus low expressors of each collagen-related gene were compared by Cox modeling and false discovery rate (FDR) corrected Mentel-Henszel test as presented in Supplementary Figure S\@ref(fig:fig-surv-uni-volcano). Unfavorable survival markers were defined by pFDR < 0.05 and hazard ratio (HR) > 1. Favorable survival markers were defined by pFDR < 0.05 and HR < 1. HR with 95% confidence intervals (CI) for unfavorable and favorable markers shared by all investigated cohorts are presented in a Forest plot._

\newpage

```{r fig-surv-uni-detail1, fig.width = figur::convert(suppl_paper_fig$surv_uni_detail1, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_uni_detail1, to = 'in')$h, fig.cap = 'Differences in biochemical relapse-free survival between high and low gene expressors: _COL1A1_, _COL3A1_, and _COL5A2_.'}

suppl_paper_fig$surv_uni_detail1$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-uni-detail1). Differences in biochemical relapse-free survival between high and low gene expressors: _COL1A1_, _COL3A1_, and _COL5A2_.__ 

_Differences in biochemical relapse-free survival in high expressors versus low expressors of each collagen-related gene were compared by Cox modeling and false discovery rate (FDR) corrected Mentel-Henszel test as presented in Supplementary Figure S\@ref(fig:fig-surv-uni-volcano). Unfavorable survival markers were defined by pFDR < 0.05 and hazard ratio (HR) > 1. Differences in biochemical relapse-free survival for the COL1A1, COL3A1, and COL5A2 genes identified as unfavorable markers in all investigated cohorts are shown in Kaplan-Meier plots. Numbers of complete cases and relapses, expression cutoff values, and numbers of high and low expressors are displayed in the plot captions. HR with 95% confidence intervals and pFDR obtained in Mentel-Henszel test are indicated in the plots._

\newpage

```{r fig-surv-uni-detail2, fig.width = figur::convert(suppl_paper_fig$surv_uni_detail2, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_uni_detail2, to = 'in')$h, fig.cap = 'Differences in biochemical relapse-free survival between high and low gene expressors: _COL11A1_, _COL11A2_, _LOXL1_, and _SERPINH1_.'}

suppl_paper_fig$surv_uni_detail2$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-uni-detail2). Differences in biochemical relapse-free survival between high and low gene expressors: _COL11A1_, _COL11A2_, _LOXL1_, and _SERPINH1_.__ 

_Differences in biochemical relapse-free survival in high expressors versus low expressors of each collagen-related gene were compared by Cox modeling and false discovery rate (FDR) corrected Mentel-Henszel test as presented in Supplementary Figure S\@ref(fig:fig-surv-uni-volcano). Unfavorable survival markers were defined by pFDR < 0.05 and hazard ratio (HR) > 1. Differences in biochemical relapse-free survival for the COL11A1, COL11A2, LOXL1, and SERPINH1 genes identified as unfavorable markers in all investigated cohorts are shown in Kaplan-Meier plots. Numbers of complete cases and relapses, expression cutoff values, and numbers of high and low expressors are displayed in the plot captions. HR with 95% confidence intervals and pFDR obtained in Mentel-Henszel test are indicated in the plots._

\newpage

```{r fig-surv-uni-detal3, fig.width = figur::convert(suppl_paper_fig$surv_uni_detal3, to = 'in')$w, fig.height = figur::convert(suppl_paper_fig$surv_uni_detal3, to = 'in')$h, fig.cap = 'Differences in biochemical relapse-free survival between high and low gene expressors: _COL4A6_, _DST_, _LAMB3_, and _PCOLCE2_.'}

suppl_paper_fig$surv_uni_detal3$plot

```

__Supplementary Figure S\@ref(fig:fig-surv-uni-detal3). Differences in biochemical relapse-free survival between high and low gene expressors: _COL4A6_, _DST_, _LAMB3_, and _PCOLCE2_.__ 

_Differences in biochemical relapse-free survival in high expressors versus low expressors of each collagen-related gene were compared by Cox modeling and false discovery rate (FDR) corrected Mentel-Henszel test as presented in Supplementary Figure S\@ref(fig:fig-surv-uni-volcano). Unfavorable survival markers were defined by pFDR < 0.05 and hazard ratio (HR) > 1. Differences in biochemical relapse-free survival for the COL4A6, DST, LAMB3, and PCOLCE2 genes identified as favorable markers in all investigated cohorts are shown in Kaplan-Meier plots. Numbers of complete cases and relapses, expression cutoff values, and numbers of high and low expressors are displayed in the plot captions. HR with 95% confidence intervals and pFDR obtained in Mentel-Henszel test are indicated in the plots._

\newpage

# References