---
title: "Prognostic and biologic relevance of collagen biosynthesis pathway in prostate cancer"
subtitle: "Supplementary Material"
author: "Urology study team"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx
    
bibliography: coll_biblio.bib
csl: frontiers-in-immunology.csl
header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "png", 
                      dpi = 600, 
                      tab.cap.pre = 'Supplementary Table ', 
                      tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', font.size = 10)


```

\newpage

# Supplementary Methods

## Data analysis software

Bioinformatic and statistical data analysis was done with R version 4.2.3. 
Basic data transformation tasks were accomplished with the _tidyverse_ package bundle [@Wickham2019] and the packages the _rlang_ [@Henry2022] and [_trafo_](https://github.com/PiotrTymoszuk/trafo).
Results were visualized using the _ggplot2_ package [@Wickham2016]. 
Venn plots were generated with _ggvenn_ package [@Yan2021]. 
Volcano and Forest plots were generated using the development package [_microViz_](https://github.com/PiotrTymoszuk/microViz). 
Ribbon and violin graphs were created with the development package [_ExDA_](https://github.com/PiotrTymoszuk/ExDA). 
Kaplan-Meier plots were generated with the _survminer_ package [@Kassambara2016] and the development packages [_kmOptimizer_](https://github.com/PiotrTymoszuk/kmOptimizer) and [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions).
Heat map visualizations of clusters and clustering diagnostic plots were generated with the package _factoextra_ [@Kassambara2020] and the development package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools).

Figures were produced by the _cowplot_ package [@Wilke2019], tables were created with the _flextable_ package [@Gohel2022].
The manuscript and the supplementary material files were written in the _rmarkdown_ environment [@Allaire2022] and rendered as Word or PDF files using the _knitr_ [@Xie2022] and _bookdown_ [@Xie2016] packages together with the _author-info-blocks.lua_ (https://github.com/pandoc/lua-filters/blob/master/author-info-blocks/author-info-blocks.lua) and _scholarly-metadata.lua_ (https://github.com/pandoc/lua-filters/blob/master/scholarly-metadata/scholarly-metadata.lua) LaTEX filters authored by Albert Krewinkel and Jörn Krenzer. 
Figures, tables R expressions and links for the R markdown documents were managed with [_figur_](https://github.com/PiotrTymoszuk/figur).

## Study cohorts and genes of interest

The following cohorts investigated in our previous report [@Eigentler2020] were re-analyzed: 

* The Cancer Gene Atlas data set [@Abeshouse2015] (n = `r study_data$tcga$expression %>% filter(tissue == 'tumor') %>% nrow` cancer samples, n = `r study_data$tcga$expression %>% filter(tissue == 'benign') %>% nrow` matched normal prostate tissue samples). This included clinical, overall and biochemical relapse-free survival data and normalized next generation sequencing counts (level 3 data) downloaded with the TCGA Assembler 2 script (https://github.com/compgenome365/TCGA-Assembler-2/blob/master/TCGA-Assembler/).

* GSE16560 data set [@Sboner2010] including normalized microarray transcriptome data, clinical and overall survival information deposited at Gene Expression Omnibus (GEO) (n = `r study_data$GSE16560$expression %>% nrow` cancer samples). The data set was fetched from GEO and imported in R with an in-house developed script.

* GSE40272 data set [@Gulzar2013] including normalized microarray transcriptome data, clinical and biochemical relapse-free survival information deposited at GEO (n = `r study_data$GSE40272$expression %>% filter(tissue == 'tumor') %>% nrow` cancer samples and n = `r study_data$GSE40272$expression %>% filter(tissue == 'benign') %>% nrow` matched normal prostate tissue samples). The data set was fetched from GEO and imported in R with an in-house developed script.

* GSE70768 data set [@Ross-Adams2015] including normalized microarray transcriptome data, clinical and biochemical relapse-free survival information deposited at GEO (n = `r study_data$GSE70768$expression %>% filter(tissue == 'tumor') %>% nrow` cancer samples and n = `r study_data$GSE70768$expression %>% filter(tissue == 'benign') %>% nrow` matched normal prostate tissue samples). The data set was fetched from GEO and imported in R with an in-house developed script.

* GSE70769 data set [@Ross-Adams2015] including normalized microarray transcriptome data, clinical and biochemical relapse-free survival information deposited at GEO (n = `r study_data$GSE70769$expression %>% nrow` cancer samples). The data set was fetched from GEO and imported in R with an in-house developed script.

The clinical information provided by the study authors included age, blood PSA concentration at diagnosis,
Gleason scoring and pathological tumor, node and metastasis staging. 
A detailed characteristic of the study cohorts is provided in __Table 1__ and by Eigentler et al. [@Eigentler2020].
Normalized, study author-provided gene expression values were transformed with the $log_2(Expression)$ and $log_2(Expression + 1)$ functions for the microarray and next generation sequencing studies, respectively. 
Gene symbol: Entrez ID conversion was accomplished with the packages _AnnotationDbi_ [@Pages2022] and _org.Hs.eg.db_ [@Carlson2022].

The list of collagen pathway genes, i.e. genes linked to collagen synthesis, post-transcriptional modification
and degradation was based on the published Collagen Signature by Kocher et al. [@Kocher2021] and the Reactome ‘Collagen
formation’ pathway ([R-HSA-1474290](https://reactome.org/content/detail/R-HSA-1474290)). 
The gene list was constrained to the set of `r nrow(globals$genes_interest)` genes detected in all studied cohorts. 
The collagen pathway genes are listed in __Table 2__.

## Non-malignant cell content in the tumor samples

Levels of non-malignant cells in the tumor samples were assessed with the MCP counter (T cells, CD8^+^ T cells, NK cells, B cells, monocytes, macrophages, myeloid dendritic cells, neutrophils, endothelial cells and cancer-associated fibroblasts) [@Becht2016] and the xCell algorithms (CD4^+^ non-regulatory T cells, CD8^+^ T cells, endothelial cells, cancer-associated fibroblasts, M1 macrophages, M2 macrophages, monocytes, neutrophils, NK cells, NKT cells, gamma-delta T cells, CD4^+^ regulatory T cells) [@Aran2017]. 
The infiltration estimates were computed using the _immunodeconv_ package [@Sturm2019].

## Gene expression in the normal prostate and prostate cancer

Expression of the collagen pathway genes was compared between the matched prostate cancer and normal prostate tissue samples in the TCGA, GSE40272 and GSE70768 cohorts by two-tailed paired T test corrected for multiple testing with Benjamini-Hochberg method [@Benjamini1995] (package [_microViz_](https://github.com/PiotrTymoszuk/microViz)).

## Correlation analysis

Pairwise correlation of the collagen pathway gene expression levels in the tumor tissue was investigated by the Pearson's correlation test. 
Correlation of the collagen score values with the Gleason scores was analyzed by the Spearman's correlation test (package [_ExDA_](https://github.com/PiotrTymoszuk/ExDA)). 
The p values were corrected for multiple testing with Benjamini-Hochberg method [@Benjamini1995]. 

## Semi-supervised clustering

Prostate cancer tissue samples were clustered in respect to normalized (Z-score) expression levels of the collagen pathway genes of interest (__Table 2__). 
Inter-cohort variability (batch effects) between the expression values was reduced by with the `ComBat()` algorithm (package _sva_) [@Leek2012]. 
Clustering of tumor samples the training TCGA cohort was accomplished by the PAM (partition around medoids) algorithm with the Manhattan distance measure (packages _cluster_ and _philentropy_, development package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)) [@Drost2018; @Schubert2019]. 
This procedure was characterized by a good explanatory performance measured by the 'explained' clustering variance statistic (ratio of the between-cluster sum of squares to the total sum of squares) and the superior stability measured as the rate of correct cluster assignment in 10-fold cross-validation (package [_clustTools_](https://github.com/PiotrTymoszuk/clustTools)) [@Lange2004; @Leng2014] in a comparison of several clustering algorithms and distance measures (__Supplementary Figure S\@ref(fig:fig-clust-dev)A__).
The number of clusters, n = 3, was chosen based on the bend of the within-cluster sum of square curve and the peak of the mean silhouette statistic (__Supplementary Figure S\@ref(fig:fig-clust-dev)B__). 
The clusters were further referred to as 'collagen clusters' and were named according to the increasing expression of the collagen subunit genes between Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters. 
Assignment of tumor samples from the test collectives (GSE16560, GSE40272, GSE70768 and GSE70769) to the collagen clusters defined in the training TCGA cohort was done by the k-NN classifier (k-nearest neighbors, inverse distance weighted, 
GSE16560: k = `r coll_clust$semi_clust$GSE16560$kNN`, GSE40272: k = `r coll_clust$semi_clust$GSE40272$kNN`, GSE70768: k = `r coll_clust$semi_clust$GSE70768$kNN`, GSE70769: k = `r coll_clust$semi_clust$GSE70769$kNN`; package _clustTools_) [@Leng2014]. 
The quality of semi-supervised clustering was assessed by comparing the distribution of the collagen clusters and the 'explained' clustering variance measure described above between the training and the test cohorts (__Supplementary Figure S\@ref(fig:fig-clust-dev)C__).

## Characteristic of the collagen clusters

Since levels of some infiltration estimates were strongly non-normally distributed, the infiltration estimates between the clusters were compared by Kruskal-Wallis test with $\eta^2$ effect size statistic. 
Gleason score and the collagen score distribution in the collagen clusters was investigated by Kruskal-Wallis test and one-way ANOVA, respectively, with $\eta^2$ effect size statistic. 
The analyses were done with the _rstatix_ package [@Kassambara2021] and the development package [_ExDA_](https://github.com/PiotrTymoszuk/ExDA)). 
P values were corrected for multiple testing with Benjamini-Hochberg method [@Benjamini1995].

## Differential gene expression

Genes differentially expressed between the collagen clusters were identified by one-way ANOVA and linear modeling with Benjamini-Hochberg correction for multiple testing (FDR: false discovery rate, package [_microViz_](https://github.com/PiotrTymoszuk/microViz)) [@Benjamini1995]. 
In brief, the transcripts significantly regulated in the Collagen^intermediate^ versus Collagen^low^ clusters (__Supplementary Table S\@ref(tab:tab-dge-int)__) and the transcripts significantly regulated in the Collagen^high^ versus Collagen^low^ clusters (__Supplementary Table S\@ref(tab:tab-dge-hi)__) were identified by the following criteria: 
(1) significance in ANOVA (pFDR~ANOVA~ < 0.05) and 
(2) significant regulation versus the Collagen^low^ cluster in linear modeling (pFDR($\beta$) < 0.05). 

## GO enrichment analysis

Biological process gene ontology (GO) term enrichment in the gene set significantly regulated in the Collagen^intermediate^ versus Collagen^low^ clusters and in the gene set significantly regulated in the Collagen^high^ versus Collagen^low^ clusters was analyzed with the _goana()_ function from the _limma_ package [@Young2010]. 
Enrichment p values were corrected for multiple testing with Benjamini-Hochberg method [@Benjamini1995]. 
The magnitude of enrichment over genomic frequency for the particular $i$-th GO term was expressed as odds ratio (OR) computed with the following formula:

$$OR_i = \frac{n_{DGE, i} \times n_{total}}{n_i \times n_{DGE}}$$

where $n_{DGE, i}$ denotes the number of differentially expressed genes assigned to the $i$-th GO term, 
$n_{total}$ denotes the total number of investigated genes, 
$n_i$ denotes the total number of genes assigned to the $i$-th GO term, 
and $n_{DGE}$ represents the total number of differentially regulated genes. 

Common significantly enriched GO terms were identified as GO terms significantly enriched in at least four cohorts, with at lest weak effect size defined as OR $\geq$ 1.44 [@Cohen2013].

## Signaling pathway modulation and visualization

Modulation of KEGG-listed signaling pathways based on fold-regulation estimates of the genes significantly regulated in the Collagen^intermediate^ versus Collagen^low^ clusters and in the gene significantly regulated in the Collagen^high^ versus Collagen^low^ clusters was done with the _SPIA_ package [@Tarca2009]. 
Significantly activated or inhibited pathways were identified by the Benjamini-Hochberg-corrected combined enrichment and perturbation significance metric pGFdr < 0.05. 
The magnitude of pathway modulation versus the Collagen^low^ clusters was measured with the tA statistic [@Tarca2009].

Pooled estimates of fold-regulation of gene expression versus the Collagen^low^ cluster were computed with the inverse variance method implemented by the package _meta_ [@Balduzzi2019]. 
Based on such pooled regulation estimates, modulation of particular components of the signaling pathways related to cancer biology modulated in at least four study cohorts (focal adhesion: KEGG ID hsa04510, ECM-receptor interaction: hsa04512, regulation of actin cytoskeleton: hsa04810 and small cell lung cancer: hsa05222, pathways in cancer: hsa05200) was visualized with the _pathview_ package [@Luo2013].

## Biochemical reactions

Rules of assignment of genes to biochemical reactions were retrieved from the Recon2 human metabolism model available via the BiGG database [@King2016] and the R package _BiGGR_ [@Gavai2015]. 
Estimates of differential expression for all available genes between the Collagen clusters were calculated by linear modeling as described above. 
Estimates of biochemical pathway fold-regulation were computed by evaluation of the gene assignment rules. 
The 'gene~A~ OR gene~B~' operator was interpreted as arithmetic mean of gene A and gene B estimates of differential expression. 
The 'gene~A~ AND gene~B~' operator was interpreted as minimum of the gene A and gene B estimates of differential expression [@Gavai2015]. 
Standard deviation, 95% confidence intervals and p values for the predicted reaction regulation estimates were obtained by a Monte Carlo simulation with n = 1000 draws from normal distribution of gene expression regulation estimates (mean: model estimate, standard deviation: standard error of the model estimate) [@Gavai2015; @Pichler2023]. 
The analysis was done with the package _BiGGR_ [@Gavai2015] and the development package [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra) [@Pichler2023]. 
The results of biochemical reaction regulation prediction for reactions found significantly regulated in all study cohorts are listed in __Supplementary Table S\@ref(tab:tab-react-int)__ and __Supplementary Table S\@ref(tab:tab-react-hi)__.

Enrichment of significantly activated or inhibited biochemical reactions within the Recon model metabolic subsystems [@King2016] was investigated by Benjamini-Hochberg-corrected Fisher's exact test (packages [_biggrExtra_](https://github.com/PiotrTymoszuk/biggrExtra) and _rstatix_) [@Kassambara2021]. 
The magnitude of enrichment was assessed with the odds ratio statistic (OR) defined for the $i$-th metabolic subsystem as follows: 

$$OR_i = \frac{n_{regulated, i} \times n_{total}}{n_i \times n_{regulated}}$$

where $n_{regulated, i}$ denotes the number of significantly regulated (activated or inhibited) biochemical reaction within the $i$-th metabolic subsystem, 
$n_{total}$ denotes the total number of investigated reactions, 
$n_i$ denotes the number of metabolic reactions within the $i$-th metabolic subsystem, and 
$n_{regulated}$ is the total number of significantly regulated biochemical reactions. 

Common significantly activated or inhibited metabolic subsystems were identified as metabolic subsystems significantly activated or inhibited in at least four cohorts, with at least weak effect size defined as OR $\geq$ 1.44 [@Cohen2013].

## Development of the collagen score

The collagen score was developed based on normalized (Z-score) expression of the collagen pathway genes in the TCGA training cohort by the Elastic Net Cox modeling with biochemical relapse-free survival as a response variable (package _glmnet_) [@Friedman2010; @Simon2011; @Zou2005]. 
The optimal value of the shrinkage parameter $\lambda$ = `r signif(coll_score$opt_lambda$lambda, 2)` was found by 200-repeat 10-fold cross-validation (CV, function `cv.glmnet()`, folds generated by the _caret_ package) and corresponded to the lowest model deviance [@Kuhn2008; @Friedman2010]. 
The collagen score was defined as the linear predictor score of the Elastic Net model fitted with the optimal $\lambda$ parameter (function `glmnet()`).  
The collagen score was calculated for the training cohort and the test collectives (GSE16560, GSE40272, GSE70768 and GSE70769) with the `predict.glmnet()` method. 
The non-zero Elastic Net model coefficients fitted in the training cohort were extracted with the `coef.glmnet()` method and exponentiated to obtain hazard ratios (HR).

## Univariable and multivariable survival modeling

Association of the collagen score values with overall (GSE16560) or biochemical relapse-free survival (remaining cohorts) was investigated by univariable Cox modeling or by multivariable Cox modeling with age and Gleason score confounders (package _survival_, development package [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions)) [@Therneau2000]. 
Proportional hazard assumption was tested with the `cox.zph()` function from the package _survival_ [@Grambsch1994]. 

Harrell's concordance index (C-index) and Grafe's integrated Brier score were used as performance metrics of the Cox models [@Harrell1996; @Graf1999] and employed in comparison of the Cox models with age and Gleason score with the Cox models including the Collagen score and clinical confounders as explanatory variables. 
Calibration of the univariable collagen score survival models was tested with a variation of the Hosmer-Lemeshov test. 
The collagen score values were stratified by the score tertiles and the Cox modeling-predicted and observed survival probabilities in the tertile strata compared with the D'Agostino - Nam $\chi^2$ statistic with two degrees of freedom (strata number - 2) [@DAgostino2003]. 
The performance and calibration statistics were computed with the development package [_coxExtensions_](https://github.com/PiotrTymoszuk/coxExtensions).

## Data and code avaibalility

Publicly available data sets were analyzed. 
Cleared data sets used for analyses will be made available upon request to the corresponding author. 
The transcriptome R analysis pipeline is available from GitHub (https://github.com/PiotrTymoszuk/collagen_pca).

\newpage

# Supplementary Tables

```{r tab-dge-int, tab.cap = 'Genes differentially regulated between the Collagen intermediate and Collagen low cluster tumors. The table is available as a supplementary Excel file.'}

suppl_tbl$dge_int[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable

```

```{r tab-dge-hi, tab.cap = 'Genes differentially regulated between the Collagen high and Collagen low cluster tumors. The table is available as a supplementary Excel file.'}

suppl_tbl$dge_hi[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable

```

```{r tab-react-int, tab.cap = 'Biochemical reactions significantly regulated in Collagen intermediate versus Collagen low shared by all study cohorts. The table is available as a supplementary Excel file.'}

suppl_tbl$react_int[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable

```

```{r tab-react-hi, tab.cap = 'Biochemical reactions significantly regulated in Collagen high versus Collagen low shared by all study cohorts. The table is available as a supplementary Excel file.'}

suppl_tbl$react_hi[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable

```


\newpage

# Supplementary Figures

```{r fig-norm-tumor-up, fig.width = mm_inch(suppl_fig$norm_tumor_up$w), fig.height = mm_inch(suppl_fig$norm_tumor_up$h), fig.cap = 'Expression of ALDH18A1, PYCR1, P4HB and PPIB in the normal prostate and prostate cancer tissue.'}

suppl_fig$norm_tumor_up$plot

```

__Supplementary Figure S\@ref(fig:fig-norm-tumor-up). Expression of__ _ALDH18A1_, _PYCR1_, _P4HB_ __and__ _PPIB_ __in the normal prostate and prostate cancer tissue.__ 
_Differences in expression of the collagen pathway genes (`r nrow(globals$genes_interest)` transcripts) between the donor-matched samples of the normal prostate and prostate cancer tissue were analyzed by two-tailed paired T test in the TCGA, GSE40272 and GSE70768 cohorts. P values were corrected for multiple testing with Benjamini-Hochberg method. log~2~-transformed expression levels for ALDH18A1 (A), PYCR1 (B), P4HB (C) and PPIB (D) are presented in plots as points. Sample matching by the tissue donor is indicated by gray lines. P values are presented in the plot captions. Numbers of tissue donors are indicated under the plots._

\newpage

```{r fig-norm-tumor-down, fig.width = mm_inch(suppl_fig$norm_tumor_down$w), fig.height = mm_inch(suppl_fig$norm_tumor_down$h), fig.cap = 'Expression of LAMB3 and COL4A2 in the normal prostate and prostate cancer tissue.'}

suppl_fig$norm_tumor_down$plot

```

__Supplementary Figure S\@ref(fig:fig-norm-tumor-down). Expression of__ _LAMB3_ __and__ _COL4A2_ __in the normal prostate and prostate cancer tissue.__ 
_Differences in expression of the collagen pathway genes (`r nrow(globals$genes_interest)` transcripts) between the donor-matched samples of the normal prostate and prostate cancer tissue were analyzed by two-tailed paired T test in the TCGA, GSE40272 and GSE70768 cohorts. P values were corrected for multiple testing with Benjamini-Hochberg method. log~2~-transformed expression levels for LAMB3 (A) and COL4A2 (B) are presented in plots as points. Sample matching by the tissue donor is indicated by gray lines. P values are presented in the plot captions. Numbers of tissue donors are indicated under the plots._

\newpage

```{r fig-corr1, fig.width = mm_inch(suppl_fig$corr1$w), fig.height = mm_inch(suppl_fig$corr1$h), fig.cap = 'Pairwise correlation of collagen pathway gene expression, TCGA and GSE16560 cohorts.'}

suppl_fig$corr1$plot

```

__Supplementary Figure S\@ref(fig:fig-corr1). Pairwise correlation of collagen pathway gene expression, TCGA and GSE16560 cohorts.__ 
_Pairwise correlation of the collagen pathway genes was investigated with Pearson's correlation test. P values were corrected for multiple testing with Benjamini-Hochberg method. Correlation coefficients (r) for significantly correlated gene pairs are presented in bubble plots. Point size corresponds to the absolute value of r, point color represents r values. Numbers of complete observations are displayed in the plot caption._

\newpage

```{r fig-corr2, fig.width = mm_inch(suppl_fig$corr2$w), fig.height = mm_inch(suppl_fig$corr2$h), fig.cap = 'Pairwise correlation of collagen pathway gene expression, GSE40272 and GSE70768 cohorts.'}

suppl_fig$corr2$plot

```

__Supplementary Figure S\@ref(fig:fig-corr2). Pairwise correlation of collagen pathway gene expression, GSE40272 and GSE70768 cohorts.__ 
_Pairwise correlation of the collagen pathway genes was investigated with Pearson's correlation test. P values were corrected for multiple testing with Benjamini-Hochberg method. Correlation coefficients (r) for significantly correlated gene pairs are presented in bubble plots. Point size corresponds to the absolute value of r, point color represents r values. Numbers of complete observations are displayed in the plot caption._

\newpage

```{r fig-corr3, fig.width = mm_inch(suppl_fig$corr3$w), fig.height = mm_inch(suppl_fig$corr3$h), fig.cap = 'Pairwise correlation of collagen pathway gene expression, GSE70769 cohorts.'}

suppl_fig$corr3$plot

```

__Supplementary Figure S\@ref(fig:fig-corr3). Pairwise correlation of collagen pathway gene expression, GSE70769 cohorts.__ 
_Pairwise correlation of the collagen pathway genes was investigated with Pearson's correlation test. P values were corrected for multiple testing with Benjamini-Hochberg method. Correlation coefficients (r) for significantly correlated gene pairs are presented in bubble plots. Point size corresponds to the absolute value of r, point color represents r values. Numbers of complete observations are displayed in the plot caption._

\newpage

```{r fig-clust-dev, fig.width = mm_inch(suppl_fig$clust_dev$w), fig.height = mm_inch(suppl_fig$clust_dev$h), fig.cap = 'Semi-supervised clustering of prostate cancer samples in respect to collagen pathway gene expression.'}

suppl_fig$clust_dev$plot

```

__Supplementary Figure S\@ref(fig:fig-clust-dev). Semi-supervised clustering of prostate cancer samples in respect to collagen pathway gene expression.__ 
_Tumor samples in the TCGA training cohort were assigned with the PAM (partition around medoids) algorithm with Manhattan distance metric into three clusters: Collagen^low^, Collagen^intermediate^ and Collagen^high^. Assignment of the tumor samples from the training GSE16560, GSE40272, GSE70768 and GSE70769 collectives to the collagen clusters was accomplished by the k-NN classifier (k-nearest neighbors, k = 7)._
_(A) Comparison of performance of several clustering algorithms (PAM: partition around medoids, HCL: hierarchical clustering/Ward D2 and KMEANS) and distance metrics (Euclidean, Manhattan, sum-of-squares and cosine) in the training TCGA cohort was measured by the amount of 'explained' clustering variance (ratio of the between-cluster sum of squares to the total sum of squares) and rate of correct cluster assignment in 10-fold cross-validation (CV). Number of complete observations and the clustering factors are shown next to the plot._
_(B, C) Determination of the cluster number for PAM/Manhattan distance clustering in the TCGA cohort by the bend of the within-cluster sum of squares (WSS) (B) and the peak of the mean silhouette statistic (C)._
_(D) Performance of the PAM/Manhattan distance clustering procedure in semi-supervised clustering in the training cohort and the test collectives measured by the 'explained' clustering variance statistic. Numbers of complete observations are presented in the Y axis._

\newpage

```{r fig-clust-hm, fig.width = mm_inch(suppl_fig$clust_hm$w), fig.height = mm_inch(suppl_fig$clust_hm$h), fig.cap = 'Expression levels of collagen pathway genes in the collagen clusters.'}

suppl_fig$clust_hm$plot

```

__Supplementary Figure S\@ref(fig:fig-clust-hm). Expression levels of collagen pathway genes in the collagen clusters.__ 
_The Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters were defined by PAM/Manhattan distance semi-supervised clustering in the training TCGA cohort and the GSE16560, GSE40272, GSE70768 and GSE70769 test collectives. Normalized levels (Z-score) of the collagen pathway transcripts in the tumor samples assigned to the collagen clusters are presented as heat maps. Tile intensities correspond to Z-score values. Numbers of samples assigned to the clusters are indicated under the plots._

\newpage

```{r fig-clust-gleason, fig.width = mm_inch(suppl_fig$clust_gleason$w), fig.height = mm_inch(suppl_fig$clust_gleason$h), fig.cap = 'Distribution of Gleason scores in the collagen clusters.'}

suppl_fig$clust_gleason$plot

```

__Supplementary Figure S\@ref(fig:fig-clust-gleason). Distribution of Gleason scores in the collagen clusters.__ 
_The Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters were defined by PAM/Manhattan distance semi-supervised clustering in the training TCGA cohort and the GSE16560, GSE40272, GSE70768 and GSE70769 test collectives. Distribution of sum Gleason scores between the collagen clusters was compared by Kruskal-Wallis test with $\eta^2$ effect size statistic. P values were corrected for multiple comparisons with Benjamini-Hochberg method. Gleason score values are presented in violin plots. Points depict single tumor samples. Red diamonds with whiskers represent medians with interquartile ranges. P and effect size statistic values are indicated in  the plot captions. Numbers of samples assigned to the clusters are shown under the plots._

\newpage

```{r fig-mcp-infil, fig.width = mm_inch(suppl_fig$mcp_infil$w), fig.height = mm_inch(suppl_fig$mcp_infil$h), fig.cap = 'MCP Counter estimates of non-malignant cell content in the collagen clusters.'}

suppl_fig$mcp_infil$plot

```

__Supplementary Figure S\@ref(fig:fig-mcp-infil). MCP Counter estimates of non-malignant cell content in the collagen clusters.__ 
_Levels of non-malignant cells (CAF: cancer-associated fibroblast, EC: endothelial cells, T cells, CD8^+^ T cells, NK cells: natural killer cells, B cells, macrophages, monocytes, neutrophils and mDC: myeloid dendritic cells) in the prostate cancer samples were estimated by the MCP counter algorithm. Differences in cell content between the Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters were analyzed with Kruskal-Wallis test. P values were corrected for multiple testing with Benjamini-Hochberg method. Mean minimum/maximum normalized cell amounts in the collagen clusters are depicted as thick lines. Tinted ribbons represent the 2 $\times$ SEM intervals. P values are shown in the Y axes. Numbers of samples assigned to the clusters are indicated under the plots._

\newpage

```{r fig-infil, fig.width = mm_inch(suppl_fig$xcell_infil$w), fig.height = mm_inch(suppl_fig$xcell_infil$h), fig.cap = 'xCell estimates of cancer-associated fibroblast and enbothelial cell infiltration in the collagen clusters.'}

suppl_fig$xcell_infil$plot

```

__Supplementary Figure S\@ref(fig:fig-infil). xCell estimates of cancer-associated fibroblast and endothelial cell infiltration in the collagen clusters.__ 
_Levels of cancer-associated fibroblasts (CAF) (A) and endothelial cells (B) in the prostate cancer samples were estimated by the xCell algorithm. Differences in CAF and endothelial cell content between the Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters were analyzed with Kruskal-Wallis test with $\eta^2$ effect size statistic. P values were corrected for multiple testing with Benjamini-Hochberg method. Infiltration estimates are presented in violin plots. Points indicate single tumor samples. Red diamonds with whiskers represent medians with interquartile ranges. P and effect size statistic values are displayed in the plot captions. Numbers of samples in the clusters are indicated under the plots._

\newpage

```{r fig-dge1, fig.width = mm_inch(suppl_fig$dge1$w), fig.height = mm_inch(suppl_fig$dge1$h), fig.cap = 'Genes differentially expressed in the collagen clusters, TCGA and GSE16560 cohorts.'}

suppl_fig$dge1$plot

```

__Supplementary Figure S\@ref(fig:fig-dge1). Genes differentially expressed in the collagen clusters, TCGA and GSE16560 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Benjamini-Hochberg-corrected p values (pFDR) and log~2~ fold regulation estimates are presented as volcano plots. Each points represent a significantly regulated gene. Point color codes for regulation sign (red: upregulated, blue: downregulated). Top most significant transcripts are labeled with their names. Numbers of samples assigned to the clusters are presented in the plot captions. Numbers of regulated transcripts are indicated under the plots._

\newpage

```{r fig-dge2, fig.width = mm_inch(suppl_fig$dge2$w), fig.height = mm_inch(suppl_fig$dge2$h), fig.cap = 'Genes differentially expressed in the collagen clusters, GSE40272 and GSE70768 cohorts.'}

suppl_fig$dge2$plot

```

__Supplementary Figure S\@ref(fig:fig-dge2). Genes differentially expressed in the collagen clusters, GSE40272 and GSE70768 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Benjamini-Hochberg-corrected p values (pFDR) and log~2~ fold regulation estimates are presented as volcano plots. Each points represent a significantly regulated gene. Point color codes for regulation sign (red: upregulated, blue: downregulated). Top most significant transcripts are labeled with their names. Numbers of samples assigned to the clusters are presented in the plot captions. Numbers of regulated transcripts are indicated under the plots._

\newpage

```{r fig-dge3, fig.width = mm_inch(suppl_fig$dge3$w), fig.height = mm_inch(suppl_fig$dge3$h), fig.cap = 'Genes differentially expressed in the collagen clusters, GSE70769 cohort.'}

suppl_fig$dge3$plot

```

__Supplementary Figure S\@ref(fig:fig-dge3). Genes differentially expressed in the collagen clusters, GSE70769 cohort.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Benjamini-Hochberg-corrected p values (pFDR) and log~2~ fold regulation estimates are presented as volcano plots. Each points represent a significantly regulated gene. Point color codes for regulation sign (red: upregulated, blue: downregulated). Top most significant transcripts are labeled with their names. Numbers of samples assigned to the clusters are presented in the plot captions. Numbers of regulated transcripts are indicated under the plots._

\newpage

```{r fig-go1, fig.width = mm_inch(suppl_fig$go1$w), fig.height = mm_inch(suppl_fig$go1$h), fig.cap = 'Top most significantly enriched biological process GO terms in the collagen clusters, TCGA, GSE16560 and GSE40272 cohorts.'}

suppl_fig$go1$plot

```

__Supplementary Figure S\@ref(fig:fig-go1). Top most significantly enriched biological process GO terms in the collagen clusters, TCGA, GSE16560 and GSE40272 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Biological process gene ontology (GO) term enrichment was investigated in the sets of significantly regulated genes with the goana tool._
_Benjamini-Hochberg corrected p values (pFDR) for the top 20 most significantly enriched GO terms in the Collagen^intermediate^ and in the Collagen^high^ tumors in the TCGA (A), GSE16560 (B) and GSE40272 (C) cohorts are presented as bar plots._

\newpage

```{r fig-go2, fig.width = mm_inch(suppl_fig$go2$w), fig.height = mm_inch(suppl_fig$go2$h), fig.cap = 'Top most significantly enriched biological process GO terms in the collagen clusters, GSE70768 and GSE70769 cohorts.'}

suppl_fig$go2$plot

```

__Supplementary Figure S\@ref(fig:fig-go2). Top most significantly enriched biological process GO terms in the collagen clusters, GSE70768 and GSE70769 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Biological process gene ontology (GO) term enrichment was investigated in the sets of significantly regulated genes with the goana tool._
_Benjamini-Hochberg corrected p values (pFDR) for the top 20 most significantly enriched GO terms in the Collagen^intermediate^ and in the Collagen^high^ tumors in the GSE70768 (A) and GSE70768 (B) cohorts are presented as bar plots._

\newpage


```{r fig-pathview1, fig.width = mm_inch(suppl_fig$pathview1$w), fig.height = mm_inch(suppl_fig$pathview1$h), fig.cap = 'Components of the focal adhesion pathway regulated in the collagen clusters.'}

suppl_fig$pathview1$plot

```

__Supplementary Figure S\@ref(fig:fig-pathview1). Components of the focal adhesion pathway regulated in the collagen clusters.__ 
_Pooled estimates for fold regulation of gene expression between the Collagen^intermediate^ and Collagen^low^ clusters (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were calculated with the inverse variance method. Modulation of components of the KEGG's focal adhesion pathway based on the common gene regulation estimates was visualized with the pathview tool. Pathway component color corresponds to the magnitude of its regulation._

\newpage

```{r fig-pathview2, fig.width = mm_inch(suppl_fig$pathview2$w), fig.height = mm_inch(suppl_fig$pathview2$h), fig.cap = 'Components of the ECM-receptor interaction pathway regulated in the collagen clusters.'}

suppl_fig$pathview2$plot

```

__Supplementary Figure S\@ref(fig:fig-pathview2). Components of the ECM-receptor interaction pathway regulated in the collagen clusters.__ 
_Pooled estimates for fold regulation of gene expression between the Collagen^intermediate^ and Collagen^low^ clusters (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were calculated with the inverse variance method. Modulation of components of the KEGG's ECM-receptor interaction pathway based on the common gene regulation estimates was visualized with the pathview tool. Pathway component color corresponds to the magnitude of its regulation._

\newpage

```{r fig-pathview3, fig.width = mm_inch(suppl_fig$pathview3$w), fig.height = mm_inch(suppl_fig$pathview3$h), fig.cap = 'Components of the actin cytoskeleton pathway regulated in the collagen clusters.'}

suppl_fig$pathview3$plot

```

__Supplementary Figure S\@ref(fig:fig-pathview3). Components of the actin cytoskeleton pathway regulated in the collagen clusters.__ 
_Pooled estimates for fold regulation of gene expression between the Collagen^intermediate^ and Collagen^low^ clusters (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were calculated with the inverse variance method. Modulation of components of the KEGG's regulation of actin cytoskeleton pathway based on the common gene regulation estimates was visualized with the pathview tool. Pathway component color corresponds to the magnitude of its regulation._

\newpage

```{r fig-pathview4, fig.width = mm_inch(suppl_fig$pathview4$w), fig.height = mm_inch(suppl_fig$pathview4$h), fig.cap = 'Components of the small cell lung cancer pathway pathway regulated in the collagen clusters.'}

suppl_fig$pathview4$plot

```

__Supplementary Figure S\@ref(fig:fig-pathview4). Components of the small cell lung cancer pathway pathway regulated in the collagen clusters.__ 
_Pooled estimates for fold regulation of gene expression between the Collagen^high^ and Collagen^low^ clusters were calculated with the inverse variance method. Modulation of components of the KEGG's small cell lung cancer pathway based on the common gene regulation estimates was visualized with the pathview tool. Pathway component color corresponds to the magnitude of its regulation._

\newpage

\newpage

```{r fig-spia1, fig.width = mm_inch(suppl_fig$spia1$w), fig.height = mm_inch(suppl_fig$spia1$h), fig.cap = 'Modulation of signaling pathways in the collagen clusters, TCGA and GSE16560 cohorts.'}

suppl_fig$spia1$plot

```

__Supplementary Figure S\@ref(fig:fig-spia1). Modulation of signaling pathways in the collagen clusters, TCGA and GSE16560 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Signaling pathway modulation based on fold regulation estimates of the significantly regulated genes in the TCGA (A) and GSE16560 cohorts (B) was investigated with the SPIA tool._
_Benjamini-Hochberg-corrected p values (pFDR) and estimates of the pathway regulation (tA) versus the Collagen^low^ cluster are presented as point plots. Point color codes for the pathway activation status (red: activated, blue: inhibited, gray: not significant). Top most significantly modulated pathways are labeled with their names. Numbers of significantly activated and inhibited pathways are displayed under the plots._

\newpage

```{r fig-spia2, fig.width = mm_inch(suppl_fig$spia2$w), fig.height = mm_inch(suppl_fig$spia2$h), fig.cap = 'Modulation of signaling pathways in the collagen clusters, GSE40272 and GSE70768 cohorts.'}

suppl_fig$spia2$plot

```

__Supplementary Figure S\@ref(fig:fig-spia2). Modulation of signaling pathways in the collagen clusters, GSE40272 and GSE70768 cohorts.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Signaling pathway modulation based on fold regulation estimates of the significantly regulated genes in the GSE40272 (A) and GSE70768 cohorts (B) was investigated with the SPIA tool._
_Benjamini-Hochberg-corrected p values (pFDR) and estimates of the pathway regulation (tA) versus the Collagen^low^ cluster are presented as point plots. Point color codes for the pathway activation status (red: activated, blue: inhibited, gray: not significant). Top most significantly modulated pathways are labeled with their names. Numbers of significantly activated and inhibited pathways are displayed under the plots._

\newpage

```{r fig-spia3, fig.width = mm_inch(suppl_fig$spia3$w), fig.height = mm_inch(suppl_fig$spia3$h), fig.cap = 'Modulation of signaling pathways in the collagen clusters, GSE70769 cohort.'}

suppl_fig$spia3$plot

```

__Supplementary Figure S\@ref(fig:fig-spia3). Modulation of signaling pathways in the collagen clusters, GSE70769 cohort.__ 
_Genes differentially regulated between the Collagen^intermediate^ and Collagen^low^ clusters and between the Collagen^high^ and Collagen^low^ clusters were identified by Benjamini-Hochberg-corrected one-way ANOVA and linear modeling. Signaling pathway modulation based on fold regulation estimates of the significantly regulated genes in the GSE70769 cohort was investigated with the SPIA tool._
_Benjamini-Hochberg-corrected p values (pFDR) and estimates of the pathway regulation (tA) versus the Collagen^low^ cluster are presented as point plots. Point color codes for the pathway activation status (red: activated, blue: inhibited, gray: not significant). Top most significantly modulated pathways are labeled with their names. Numbers of significantly activated and inhibited pathways are displayed under the plots._

\newpage

```{r fig-react-int, fig.width = mm_inch(suppl_fig$react_int$w), fig.height = mm_inch(suppl_fig$react_int$h), fig.cap = 'Differences in predicted biochemical reaction activity between Collagen intermediate and Collagen low cancers.'}

suppl_fig$react_int$plot

```

__Supplementary Figure S\@ref(fig:fig-react-int). Differences in predicted biochemical reaction activity between Collagen intermediate and Collagen low cancers.__ 
_Estimates of biochemical reaction modulation between the Collagen^intermediate^ and Collagen^low^ clusters were computed by evaluating gene association rules in the Recon2 human metabolism model with the BiGGR and biggrExtra tools. Pooled reaction regulation estimates and their 95% confidence interval for reactions found significantly modulated in all study cohorts were calculated with the inverse variance method. The pooled regulation estimates with 95% confidence intervals are shown in a Forest plot. Reactions are labeled with their BiGG database identifiers and were classified as members of aminoacid metabolism (AA), bile acid synthesis (BAS), metabolism of extracellular matrix glycoproteins (ECM), eicosanoid metabolism (ECSM), energy metabolism (Energy), fatty acid metabolism (FAM), fatty acid oxidation (FAOx) and aminoacid amine group metabolism (Urea).__

\newpage

```{r fig-react-hi, fig.width = mm_inch(suppl_fig$react_hi$w), fig.height = mm_inch(suppl_fig$react_hi$h), fig.cap = 'Differences in predicted biochemical reaction activity between Collagen high and Collagen low cancers.'}

suppl_fig$react_hi$plot

```

__Supplementary Figure S\@ref(fig:fig-react-hi). Differences in predicted biochemical reaction activity between Collagen high and Collagen low cancers.__ 
_Estimates of biochemical reaction modulation between the Collagen^high^ and Collagen^low^ clusters were computed by evaluating gene association rules in the Recon2 human metabolism model with the BiGGR and biggrExtra tools. Pooled reaction regulation estimates and their 95% confidence interval for reactions found significantly modulated in all study cohorts were calculated with the inverse variance method. The pooled regulation estimates with 95% confidence intervals are shown in a Forest plot. Reactions are labeled with their BiGG database identifiers and were classified as members of aminoacid metabolism (AA), bile acid synthesis (BAS), metabolism of extracellular matrix glycoproteins (ECM), eicosanoid metabolism (ECSM), energy metabolism (Energy), fatty acid metabolism (FAM), fatty acid oxidation (FAOx) and aminoacid amine group metabolism (Urea)._

\newpage

```{r fig-faox, fig.width = mm_inch(suppl_fig$faox$w), fig.height = mm_inch(suppl_fig$faox$h), fig.cap = 'Predicted regulation of fatty acid oxidation reactions in the Collagen clusters.'}

suppl_fig$faox$plot

```

__Supplementary Figure S\@ref(fig:fig-faox). Predicted regulation of fatty acid oxidation reactions in the Collagen clusters.__ 
_Estimates of biochemical reaction modulation between the Collagen^intermediate^ and Collagen^low^ (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were computed by evaluating gene association rules in the Recon2 human metabolism model with the BiGGR and biggrExtra tools. Predicted regulation estimates for `r nrow(meta_hg$faox_regulation_int[[1]])` reactions involved in fatty acid oxidation (FAOx) are presented as bars. Bar colors code for regulation sign and significance. Numbers of significantly activated and inhibited reactions are displayed in the graph captions._

\newpage

```{r fig-tca, fig.width = mm_inch(suppl_fig$tca$w), fig.height = mm_inch(suppl_fig$tca$h), fig.cap = 'Predicted regulation of TCA cycle reactions in the Collagen clusters.'}

suppl_fig$tca$plot

```

__Supplementary Figure S\@ref(fig:fig-tca). Predicted regulation of TCA cycle reactions in the Collagen clusters.__ 
_Estimates of biochemical reaction modulation between the Collagen^intermediate^ and Collagen^low^ (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were computed by evaluating gene association rules in the Recon2 human metabolism model with the BiGGR and biggrExtra tools. Predicted regulation estimates for the TCA (Tri-Carboxylic Acid) cycle reactions are presented as bars. Bar colors code for regulation sign and significance. The reactions are labeled with their BiGG database IDs._

\newpage

```{r fig-oxphos, fig.width = mm_inch(suppl_fig$oxphos$w), fig.height = mm_inch(suppl_fig$oxphos$h), fig.cap = 'Predicted regulation of oxidative phosphorylation reactions in the Collagen clusters.'}

suppl_fig$oxphos$plot

```

__Supplementary Figure S\@ref(fig:fig-oxphos). Predicted regulation of oxidative phosphorylation reactions in the Collagen clusters.__ 
_Estimates of biochemical reaction modulation between the Collagen^intermediate^ and Collagen^low^ (A) and between the Collagen^high^ and Collagen^low^ clusters (B) were computed by evaluating gene association rules in the Recon2 human metabolism model with the BiGGR and biggrExtra tools. Predicted regulation estimates for the oxidative phosphorylation (OxPhos) reactions are presented as bars. Bar colors code for regulation sign and significance. The reactions are labeled with the OxPhos complex names._

\newpage

```{r fig-cs-gleason, fig.width = mm_inch(suppl_fig$cs_gleason$w), fig.height = mm_inch(suppl_fig$cs_gleason$h), fig.cap = 'Association of the collagen score with the Gleason score and collagen cluster assignment.'}

suppl_fig$cs_gleason$plot

```

__Supplementary Figure S\@ref(fig:fig-cs-gleason). Association of the collagen score with the Gleason score and collagen cluster assignment.__ 
_Collagen Score based on the normalized expression levels of the collagen pathway genes was developed using the elastic net Cox modeling algorithm._
_(A) Correlation of the collagen score values with the sum Gleason score was investigated by Spearman's correlation test. P values were corrected for multiple comparisons by Benjamini-Hochberg method. Collagen score and Gleason score values are presented as points. The points were jittered horizontally to reduce overplotting. Blue lines with gray ribbons represent linear trends with 95% confidence regions. Correlation coeffcients ($\rho$) with 95% confidence intervals, p values and numbers of complete observations are displayed in the plot captions._
_(B) Levels of the collagen score were compared between the Collagen^low^, Collagen^intermediate^ and Collagen^high^ clusters by one-way ANOVA with $\eta^2$ effect size statistic. P values were corrected for multiple comparisons with Benjamini-Hochberg method. Collagen score values are presented in violin plots. Points depict single tumor samples. Red diamonds and whiskers represent medians with interquartile ranges. P and effect size statistic values are shown in the plot captions. Numbers of samples assigned to the clusters are displayed under the plots._

\newpage

```{r fig-cs-multi, fig.width = mm_inch(suppl_fig$cs_multi$w), fig.height = mm_inch(suppl_fig$cs_multi$h), fig.cap = 'Prediction of prostate cancer survival by the collagen score with Gleason score and age confounder variables.'}

suppl_fig$cs_multi$plot

```

__Supplementary Figure S\@ref(fig:fig-cs-multi). Prediction of prostate cancer survival by the collagen score with Gleason score and age confounder variables.__ 
_Association of the collagen score with biochemical relapse-free (RFS: TCGA, GSE40272, GSE70768, GSE70769) or overall survival (OS: GSE16560) were investigated by uni-variable Cox proportional hazard regression or by multi-variable Cox modeling with age and Gleason score as confounders. Hazard ratio of the collagen score estimates with 95% confidence intervals for the uni- and multi-variable models are presented in a Forest plot. Survival response type and numbers of complete observations are indicated in the Y axis._

\newpage

# References